\documentclass[]{article}
\usepackage[backref,bookmarks=true,bookmarksopen=true,pdftitle=SoftInstall,pdfauthor=YU Zhejian]{hyperref}
\renewcommand{\sfdefault}{phv}
\renewcommand{\rmdefault}{phv}
\usepackage{indentfirst}
\renewcommand\familydefault{\sfdefault}
\usepackage[inner=0.5in,outer=0.5in]{geometry}
\usepackage{titletoc}
\titlecontents{part}[2em]{\addvspace{4pt}\large}{\contentslabel[\thecontentslabel]{4em}}{}{\ \titlerule*[.5pc]{.}\ \thecontentspage}[]
\titlecontents{section}[1em]{\addvspace{2pt}}{\bfseries\contentslabel[\thecontentslabel]{4em}}{}{\ \titlerule*[.5pc]{.}\ \thecontentspage}[]

\begin{document}
\title{\fontsize{40}{40}\selectfont\scshape Fantastic Software\\{\Huge and}\\Where to Find Them}
\author{YU Zhejian}
\maketitle
\setcounter{section}{-1}
\section{Introduction}

This book involves the software used in the common bioinformatics pipelines and the way to install them.

The term ``\textbf{deprecated}'' means that the entire software was deprecated by its developer, while ``\textbf{branch deprecated}'' means that although this branch was deprecated, the software is still being developed somewhere else. Unless there is \textbf{EXPLICIT} information, all branches \& software are considered not deprecated.

The term ``pre-build binary'' means 1. Compiled binaries for C, C++ or D, etc. or 2. Compiled documentations for \LaTeX or asciidoc, etc. or 3. Scripts for scripting languages like Perl, Shell or Python, etc. Those which is not ``pre-build binary'' are compiled locally by ourselves. ``N/A'' means we have not, or failed to install it.

\tableofcontents

\part{Read Data Pre-Processing}

\section{BamTools}

Sourceforge: \url{https://sourceforge.net/projects/bamtools/}, deprecated

GitHub: \url{https://github.com/pezmaster31/bamtools}, v2.5.1, 2019-12-24

Current: GitHub, self-built biniary, v2.5.1

License: MIT

C++ 96.4\% Python 1.8\% CMake 1.3\% Other 0.5\%

Barnett, D. W., Garrison, E. K., Quinlan, A. R., Strömberg, M. P., \& Marth, G. T. (2011). BamTools: a C++ API and toolkit for analyzing and managing BAM files. Bioinformatics, 27(12), 1691–1692. \url{https://doi.org/10.1093/bioinformatics/btr174}

Updated: 2020-06-25

\section{htsjdk}

GitHub: \url{https://github.com/samtools/htsjdk}, v2.21.3 in 2020-03-10

Official: \url{http://samtools.github.io/htsjdk/}

Current: N/A

Java 99.8\% Other 0.2\% 

Updated: 2020-06-10

\section{htslib}

GitHub: \url{https://github.com/samtools/htslib} with MIT/Expat license, Modified 3-Clause BSD license, v1.10.2 in 2019-12-21 

Conda: \url{https://anaconda.org/bioconda/htslib} with MIT License, v1.10.2 in 2019-12-23

Official: \url{http://www.htslib.org/}

Current: GitHub, self-build,  1.10.2-23-g6b72368

C 89.3\% C++ 3.7\% Perl 2.7\% Makefile 1.4\% Roff 1.2\% M4 1.0\% Other 0.7\%

Updated: 2020-06-10

\section{SeqAn2}

Official: \url{https://www.seqan.de/}

GitHub: \url{https://github.com/seqan/seqan}, v2.4.0 in 2018-02-03

Current: GitHub, master

License: The 3-Clause BSD License

C++ 63.9\% PHP 22.6\% Python 3.4\% HTML 2.7\% JavaScript 2.3\% C 1.7\% Other 3.4\%

Döring, A., Weese, D., Rausch, T., \& Reinert, K. (2008). SeqAn an efficient, generic C++ library for sequence analysis. BMC Bioinformatics, 9. \url{https://doi.org/10.1186/1471-2105-9-11}

Updated: 2020-06-25

\section{SeqAn3}

Official: \url{https://www.seqan.de/}

GitHub: \url{https://github.com/seqan/seqan3}, v3.0.1 in 2020-01-22

Current: GitHub, master

License: The 3-Clause BSD License

C++ 98.9\% CMake 1.1\%

Updated: 2020-06-25

\section{SRA-Toolkit}

Conda: \url{https://anaconda.org/bioconda/sra-tools}, v2.10.7 in 2020-06-03

Official: \url{https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software}

GitHub: \url{https://github.com/ncbi/sra-tools/}, v2.10.7 in 2020-05-22

Current: GitHub, pre-build binary, 2.10.4

License: PUBLIC DOMAIN NOTICE

C 70.2\% C++ 19.4\% Makefile 4.0\% Shell 2.3\% Perl 1.9\ Python 1.0\% Other 1.2\%

Updated: 2020-06-10

\section{Tabix (deprecated)}

Conda: \url{https://anaconda.org/bioconda/tabix}, v0.2.6 in 2018-10

GitHub: \url{https://github.com/samtools/tabix}

SourceForge: \url{https://sourceforge.net/projects/samtools}, branch deprecated

Current: N/A

License: BSD

C 83.8\% Java 6.8\% Python 4.6\% TeX 2.8\% Perl 2.0\%

Updated: 2020-06-10

\part{Read Quality Controller}

\section{CutAdapt}

Conda: \url{https://anaconda.org/bioconda/cutadapt}, v2.10 in 2020-04-23

GitHub: \url{https://github.com/marcelm/cutadapt}, v2.10 in 2020-04-22

Official: \url{https://cutadapt.readthedocs.io/en/stable/}

CutAdapt: Conda, 2.10

License: MIT

Python 99.6\% Shell 0.4\%

Martin, M. (2011). Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet.Journal, 17(1), 10. \url{https://doi.org/10.14806/ej.17.1.200}

Updated: 2020-06-10

\section{FastQC}

Conda: \url{https://anaconda.org/bioconda/fastqc} with GPL 3, v0.11.9 in 2020-02-23

Official: \url{http://www.bioinformatics.babraham.ac.uk/projects/fastqc/} with GPL 3

GitHub: \url{https://github.com/s-andrews/FastQC} with GPL 2, v0.11.9 in 2020-01-08

Current: GitHub, pre-build binary, 0.11.9

Java 95.5\% HTML 3.1\% Perl 1.2\% Other 0.2\%

Updated: 2020-06-10

\section{QualiMap}

Official: \url{http://qualimap.bioinfo.cipf.es/}

BitBucket: \url{https://bitbucket.org/kokonech/qualimap/}, build 11-11-19 in 2019-11-11

Current: N/A

License: GPL 2

Java

García-Alcalde, F., Okonechnikov, K., Carbonell, J., Cruz, L. M., Götz, S., Tarazona, S., Dopazo, J., Meyer, T. F. and Conesa, A. (2012) Qualimap: evaluating next-generation sequencing alignment data, Bioinformatics, 28(20), pp. 2678–2679. doi: 10.1093/bioinformatics/bts503.

Okonechnikov, K., Conesa, A. and García-Alcalde, F. (2015) Qualimap 2: advanced multi-sample quality control for high-throughput sequencing data, Bioinformatics, 32(2), pp. 292–294. doi: 10.1093/bioinformatics/btv566.

Updated: 2020-06-10

\section{Trimmomatic}

Official: \url{http://www.usadellab.org/cms/index.php?page=trimmomatic}

GitHub: \url{https://github.com/timflutre/trimmomatic}

Conda: \url{https://anaconda.org/bioconda/trimmomatic}, v0.39 in 2019-11-21

Current: GitHub, pre-build binary, v0.39

License: GPL 3

Java 99.2\% Makefile 0.8\%

Bolger, A. M., Lohse, M., \& Usadel, B. (2014). Trimmomatic: A flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114–2120.\url{https://doi.org/10.1093/bioinformatics/btu170}

Updated: 2020-06-10

\part{Aligner}
\section{Bowtie}

Official: \url{http://bowtie-bio.sourceforge.net/index.shtml}, v1.2.3, 2019-07-05

GitHub: \url{https://github.com/BenLangmead/bowtie}, v1.2.3 in 2019-07-06

Conda: \url{https://anaconda.org/bioconda/bowtie}, v1.2.3 in 2020-05-09

Current: GitHub, pre-build binary

License: Artistic License 2.0

C++ 91.1\% Perl 6.2\% Shell 0.9\% C 0.7\% Python 0.6\% Makefile 0.4\% Objective-C 0.1\%

Langmead, B., Trapnell, C., Pop, M., \& Salzberg, S. L. (2009). Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biology, 10(3), R25. \url{https://doi.org/10.1186/gb-2009-10-3-r25}Langmead, B., \& Salzberg, S. L. (2012).

\section{Bowtie2}

Conda: \url{https://anaconda.org/bioconda/bowtie2}, v2.4.1 in 2020-06-07

Official: \url{http://bowtie-bio.sourceforge.net/bowtie2/index.shtml}, v2.4.1 in 2020-02-28

GitHub: \url{https://github.com/BenLangmead/bowtie2}, v2.4.1 in 2020-02-29

Current: GitHub, pre-build binary

License: GPL 3

C++ 80.7\% Perl 14.5\% Python 1.5\% C 1.3\% Shell 1.1\% Makefile 0.6\% CMake 0.3\%

Fast gapped-read alignment with Bowtie 2. Nature Methods, 9(4), 357–359. \url{https://doi.org/10.1038/nmeth.1923}

Updated: 2020-06-10


\section{BWA}

Conda: \url{https://anaconda.org/bioconda/bwa}, v0.7.17 in 2019-12-26

GitHub: \url{https://github.com/lh3/bwa}, v0.7.17-r1188 in 2017-10-24

Official: \url{http://bio-bwa.sourceforge.net/}

SourceForge: \url{https://sourceforge.net/projects/bio-bwa/}, v0.7.17 in 2017

Current: GitHub, self-build, v0.7.17-r1188

License: GPL 3

C 85.7\% JavaScript 5.8\% Roff 4.3\% C++ 1.8\% Perl 1.3\% Shell 0.6\% Makefile 0.5\%

\textbf{BWA-backtrack}

Li, H., \& Durbin, R. (2009). Fast and accurate short read alignment with Burrows-Wheeler transform. Bioinformatics, 25(14), 1754–1760. \url{https://doi.org/10.1093/bioinformatics/btp324}

\textbf{BWA-SW}

Li, H., \& Durbin, R. (2010). Fast and accurate long-read alignment with Burrows-Wheeler transform. Bioinformatics, 26(5), 589–595. \url{https://doi.org/10.1093/bioinformatics/btp698}

\textbf{BWA-MEM}

Li, H. (2013) Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. Available at: \url{http://arxiv.org/abs/1303.3997} (Accessed: 10 April 2020).

Updated: 2020-06-10
\section{HiSat2}

Conda: \url{https://anaconda.org/bioconda/hisat2}, v2.2.0 in 2020-03-02

Official: \url{https://daehwankimlab.github.io/hisat2/}, v2.2.0 in 2020-02-07

GitHub: \url{https://github.com/DaehwanKimLab/hisat2}

Current: Official, pre-build binary, 2.2.0

License: GPL 3

C++ 57.8\% C 16.9\% Python 15.9\% Perl 6.2\% Shell 1.3\% HTML 1.3\% Other 0.6\%

Langmead, B., \& Salzberg, S. L. (2012). Fast gapped-read alignment with Bowtie 2. Nature Methods, 9(4), 357–359. \url{https://doi.org/10.1038/nmeth.1923}

Updated: 2020-06-10


\section{SubRead}

Conda: \url{https://anaconda.org/bioconda/subread}, v2.0.1 in 2020-06-05

SourceForge: \url{https://sourceforge.net/projects/subread}, v2.0.1 in 2020-05-13

Official: \url{http://subread.sourceforge.net/}

Current: Official, pre-build binary, v2.0.0

License: GPL 3

Liao, Y., Smyth, G. K., \& Shi, W. (2013). The Subread aligner: Fast, accurate and scalable read mapping by seed-and-vote. Nucleic Acids Research, 41(10). \url{https://doi.org/10.1093/nar/gkt214}

Liao, Y., Smyth, G. K. and Shi, W. (2019) The R package Rsubread is easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads, Nucleic acids research. NLM (Medline), 47(8), p. e47. doi: 10.1093/nar/gkz114.

Liao, Y., Smyth, G. K. and Shi, W. (2014) FeatureCounts: An efficient general purpose program for assigning sequence reads to genomic features, Bioinformatics. Oxford University Press, 30(7), pp. 923–930. doi: 10.1093/bioinformatics/btt656.

Updated: 2020-06-10

\section{TopHat2 (deprecated)}

Official: \url{http://ccb.jhu.edu/software/tophat/index.shtml}, v2.1.1 in 2020-02-23

GitHub: \url{https://github.com/DaehwanKimLab/tophat}, v2.1.2 in 2018-05-24

Conda: \url{https://anaconda.org/bioconda/tophat}, v2.1.1 in 2019-05

Current: N/A

License: BSL 1.0

C++ 84.8\% C 10.0\% Python 3.8\% Perl 0.6\% Makefile 0.3\% M4 0.2\% Other 0.3\%

Trapnell, C., Pachter, L. and Salzberg, S. L. (2009) TopHat: discovering splice junctions with RNA-Seq., Bioinformatics (Oxford, England), 25(9), pp. 1105–1111. doi: 10.1093/bioinformatics/btp120.

Langmead, B., Trapnell, C., Pop, M. and Salzberg, S. L. (2009) Ultrafast and memory-efficient alignment of short DNA sequences to the human genome, Genome Biology. BioMed Central, 10(3), p. R25. doi: 10.1186/gb-2009-10-3-r25.

Kim, D., Pertea, G., Trapnell, C., Pimentel, H., Kelley, R. and Salzberg, S. L. (2013) TopHat2: Accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions, Genome Biology. BioMed Central, 14(4), p. R36. doi: 10.1186/gb-2013-14-4-r36.

Kim, D. and Salzberg, S. L. (2011) TopHat-Fusion: An algorithm for discovery of novel fusion transcripts, Genome Biology. BioMed Central, 12(8), p. R72. doi: 10.1186/gb-2011-12-8-r72.

Updated: 2020-06-10

\part{SAM/BAM/CRAM File Processor}

\section{Picard}

Official: \url{http://broadinstitute.github.io/picard/}

Conda: \url{https://anaconda.org/bioconda/picard}, v2.22.9 in 2020-06-03

GitHub: \url{https://github.com/broadinstitute/picard}, v2.23.0 in 2020-06-10

Current: GitHub, pre-build binary, v2.21.6-SNAPSHOT

License: MIT

Java 98.7\% R 0.4\% XSLT 0.4\% Shell 0.2\% HTML 0.2\% CSS 0.1\% 

``Picard Toolkit.'' 2019. Broad Institute, GitHub Repository. http://broadinstitute.github.io/picard/; Broad Institute

Updated: 2020-06-10

\section{Sambamba}

Conda: \url{https://anaconda.org/bioconda/sambamba}, v0.7.1 in 2020-03-15 

Official: \url{http://www.sambamba.org/}

Related: \url{https://thebird.nl/blog/D_Dragon.html}

GitHub: \url{https://github.com/biod/sambamba}, v0.7.1 (20191128) in 2019-11-29

Current: GitHub, pre-build binary, 0.7.1

\begin{verbatim}
LDC 1.17.0 / DMD v2.087.1 / LLVM8.0.1 / bootstrap LDC - the LLVM D compiler
(1.17.0)
\end{verbatim}

License: GPL 2

D 82.0\% Shell 7.7\% Roff 6.0\% Python 1.6\% Ruby 1.1\% Meson 0.8\% Makefile 0.8\%

Tarasov, A., Vilella, A. J., Cuppen, E., Nijman, I. J., \& Prins, P. (2015). Sambamba: fast processing of NGS alignment formats. Bioinformatics, 31(12), 2032–2034. \url{https://doi.org/10.1093/bioinformatics/btv098}

Updated: 2020-06-10

\section{SAMBLASTER}

GitHub: \url{https://github.com/GregoryFaust/samblaster}, v0.1.26 in 2020-06-16

Conda: \url{https://anaconda.org/bioconda/samblaster}, v0.1.26 in 2020-06-16

Current: Github, self-build binary, v0.1.26

License: MIT

C++ 99.4\% Makefile 0.6\%

Faust, G. G., \& Hall, I. M. (2014). SAMBLASTER: fast duplicate marking and structural variant read extraction. Bioinformatics, 30(17), 2503–2505. \url{https://doi.org/10.1093/bioinformatics/btu314}

Updated: 2020-06-25

\section{Samtools}

Official: \url{http://www.htslib.org/}

SourceForge: \url{http://samtools.sourceforge.net/}, deprecated

Conda: \url{https://anaconda.org/bioconda/samtools}, v1.10 in 2020-04-16

GitHub: \url{https://github.com/samtools/samtools}, v1.10 in 2019-12-07

Current: GitHub, self-build, v1.10 with htslib v1.10

License: The MIT/Expat License

C 73.3\% Perl 18.8\% M4 2.7\% Lua 1.8\% Shell 1.4\% Makefile 1.0\% Other 1.0\%

\textbf{VCF Format:}

Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., … Durbin, R. (2011). The variant call format and VCFtools. Bioinformatics, 27(15), 2156–2158. \url{https://doi.org/10.1093/bioinformatics/btr330}

\textbf{SAM \& BAM Format:}

Cock, P. J. A., Bonfield, J. K., Chevreux, B., \& Li, H. (2015). SAM/BAM format v1.5 extensions for de novo assemblies. BioRxiv, 020024. \url{https://doi.org/10.1101/020024}

Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., … Durbin, R. (2009). The Sequence Alignment/Map format and SAMtools. Bioinformatics, 25(16), 2078–2079. \url{https://doi.org/10.1093/bioinformatics/btp352}

\textbf{CRAM Format:}

Fritz, M. H. Y., Leinonen, R., Cochrane, G., \& Birney, E. (2011). Efficient storage of high throughput DNA sequencing data using reference-based compression. Genome Research, 21(5), 734–740. \url{https://doi.org/10.1101/gr.114819.110}

\textbf{The original mpileup calling algorithm plus mathematical notes (mpileup/bcftools call -c)}

Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics, 27(21), 2987–2993. \url{https://doi.org/10.1093/bioinformatics/btr509}

LI, H. (2010). Mathematical Notes on SAMtools Algorithms.

\textbf{Mathematical notes for the updated multiallelic calling model (mpileup/bcftools call -m)}

Petr Danecek, Stephan Schiffels, R. D. (2016). Multiallelic calling model in bcftools (-m). Retrieved January 30, 2020, from \url{http://samtools.github.io/bcftools/call-m.pdf}

\textbf{Hidden Markov model for detecting runs of homozygosity (bcftools roh)}

Narasimhan, V., Danecek, P., Scally, A., Xue, Y., Tyler-Smith, C., \& Durbin, R. (2016). BCFtools/RoH: A hidden Markov model approach for detecting autozygosity from next-generation sequencing data. Bioinformatics, 32(11), 1749–1751. \url{https://doi.org/10.1093/bioinformatics/btw044}

\textbf{Copy number variation/aneuploidy calling from microarray data (bcftools cnv/bcftools polysomy)}


Danecek, P., McCarthy, S. A., Consortium, H. S., \& Durbin, R. (2016). A method for checking genomic integrity in cultured cell lines from snp genotyping data. PLoS ONE, 11(5). \url{https://doi.org/10.1371/journal.pone.0155014}

\textbf{Haplotype-aware calling of variant consequences (bcftools csq)}

Danecek, P., \& McCarthy, S. A. (2017). BCFtools/csq: Haplotype-aware variant consequences. Bioinformatics, 33(13), 2037–2039. \url{https://doi.org/10.1093/bioinformatics/btx100}

\textbf{Base alignment quality (BAQ) method improve SNP calling around INDELs}


Li, H. (2011). Improving SNP discovery by base alignment quality. Bioinformatics, 27(8), 1157–1158. \url{https://doi.org/10.1093/bioinformatics/btr076}

\textbf{Segregation based QC metric originally implemented in SGA}

Durbin, R. (2014). Segregation based metric for variant call QC. Retrieved from \url{http://samtools.github.io/bcftools/rd-SegBias.pdf}

Updated: 2020-06-10

\part{Variant Caller: SNP and Indel}
\section{FreeBayes}

GitHub: \url{https://github.com/ekg/freebayes}, v1.3.2 in 2019-12-17

Conda: \url{https://anaconda.org/bioconda/freebayes}, v1.3.2 in 2020-05-09

Current: GitHub, pre-build binary, v1.3.1-dirty

License: MIT

C++ 95.7\% TeX 2.1\% Python 0.9\% Shell 0.5\% C 0.4\% Assembly 0.3\% Other 0.1\%

Garrison, E., \& Marth, G. (2012). Haplotype-based variant detection from short-read sequencing. Retrieved from \url{http://arxiv.org/abs/1207.3907}

Updated: 2020-06-10


\section{GATK3 (deprecated)}

Conda: \url{https://anaconda.org/bioconda/gatk}, v3.8 in 2019-10-17

Docker: \url{https://hub.docker.com/r/broadinstitute/gatk3/}, v3.8.1 in 2014

GoogleCloud: \url{https://console.cloud.google.com/storage/browser/gatk-software/package-archive/gatk/}, v3.8-1-0-gf15c1c3ef in 2019-12-09

Current: GoogleCloud, pre-build binary, 3.8-1-0-gf15c1c3ef

Updated: 2020-06-10

\section{GATK4}

Official: \url{https://software.broadinstitute.org/gatk/}

GitHub: \url{https://github.com/broadinstitute/gatk}, v4.1.7.0 in 2020-04-24

Conda: \url{https://anaconda.org/bioconda/gatk4}, v4.1.7.0 in 2020-05-23

Docker: \url{https://hub.docker.com/r/broadinstitute/gatk/}, v4.1.7.0 in 2020-04

Current: GitHub, pre-build binary

\begin{verbatim}
Using GATK jar gatk-package-4.1.4.1-local.jar
Running:
java -Dsamjdk.use_async_io_read_samtools=false
-Dsamjdk.use_async_io_write_samtools=true
-Dsamjdk.use_async_io_write_tribble=false
-Dsamjdk.compression_level=2 -jar 
gatk-package-4.1.4.1-local.jar --version
The Genome Analysis Toolkit (GATK) v4.1.4.1
HTSJDK Version: 2.21.0
Picard Version: 2.21.2
\end{verbatim}

License: BSD-3-Clause

Java 93.8\% Python 2.8\% wdl 1.9\% Shell 1.1\% R 0.3\% HTML 0.1\%

\textbf{The fourth paper, technically just a manuscript deposited in bioRxiv -- but it counts! This is a good citation to include in a Materials and Methods section or in a Discussion if you're talking about the joint calling process.}

Poplin, R., Ruano-Rubio, V., DePristo, M. A., Fennell, T. J., Carneiro, M. O., Auwera, G. A. Van der, … Banks, E. (2017). Scaling accurate genetic variant discovery to tens of thousands of samples. BioRxiv, 201178. \url{https://doi.org/10.1101/201178}

\textbf{The first GATK paper covers the computational philosophy underlying the GATK and is a good citation for the GATK in general.}

McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., … DePristo, M. A. (2010). The genome analysis toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data. Genome Research, 20(9), 1297–1303. \url{https://doi.org/10.1101/gr.107524.110}

\textbf{The second GATK paper describes in more detail some of the key tools commonly used in the GATK for high-throughput sequencing data processing and variant discovery. The paper covers base quality score recalibration, indel realignment, SNP calling with UnifiedGenotyper, variant quality score recalibration and their application to deep whole genome, whole exome, and low-pass multi-sample calling. This is a good citation if you use the GATK for variant discovery.}

Depristo, M. A., Banks, E., Poplin, R., Garimella, K. V., Maguire, J. R., Hartl, C., … Daly, M. J. (2011). A framework for variation discovery and genotyping using next-generation DNA sequencing data. Nature Genetics, 43(5), 491–501. \url{https://doi.org/10.1038/ng.806}

\textbf{The third GATK paper describes the Best Practices for Variant Discovery (version 2.x). It is intended mainly as a learning resource for first-time users and as a protocol reference. This is a good citation to include in a Materials and Methods section.}

Van der Auwera, G. A., Carneiro, M. O., Hartl, C., Poplin, R., del Angel, G., Levy-Moonshine, A., … DePristo, M. A. (2013). From fastQ data to high-confidence variant calls: The genome analysis toolkit best practices pipeline. Current Protocols in Bioinformatics, (SUPL.43). \url{https://doi.org/10.1002/0471250953.bi1110s43}

Updated: 2020-06-10


\section{LoFreq}

SourceForge: \url{https://sourceforge.net/projects/lofreq/}, v2.1.2 in 2015-05-19

Conda: \url{https://anaconda.org/bioconda/lofreq}, v2.1.4 in 2020-05-09

Official: \url{http://csb5.github.io/lofreq/}

GitHub: \url{https://github.com/CSB5/lofreq}, v2.1.4 in 2020-01-05

Current: GitHub, pre-build binary

\begin{verbatim}
version: 2.1.4
commit: unknown
build-date: Feb 19 2020
\end{verbatim}

License: MIT License

C 74.2\% Python 17.8\% Shell 5.3\% M4 2.3\% Other 0.4\%

Wilm, A., Aw, P. P. K., Bertrand, D., Yeo, G. H. T., Ong, S. H., Wong, C. H., … Nagarajan, N. (2012). LoFreq: A sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets. Nucleic Acids Research, 40(22), 11189–11201. \url{https://doi.org/10.1093/nar/gks918}

Updated: 2020-06-10
\section{Manta}

Conda: \url{https://anaconda.org/bioconda/manta}, v1.6.0 in 2019-10-06

GitHub: \url{https://github.com/Illumina/manta}, v1.6.0 in 2019-06-25

Current: GitHub, pre-build binary, v1.6.0

License: GPL 3

C++ 87.3\% Python 7.2\% CMake 4.3\% Shell 0.8\% Dockerfile 0.3\% C 0.1\%

Chen, X., Schulz-Trieglaff, O., Shaw, R., Barnes, B., Schlesinger, F., Källberg, M., … Saunders, C. T. (2016). Manta: Rapid detection of structural variants and indels for germline and cancer sequencing applications. Bioinformatics, 32(8), 1220–1222. \url{https://doi.org/10.1093/bioinformatics/btv710}

Updated: 2020-06-10


\section{MuSE}

Official: \url{https://bioinformatics.mdanderson.org/public-software/muse/}

GitHub: \url{https://github.com/danielfan/MuSE}, v1.0-rc in 2016-07-09

Conda: \url{https://anaconda.org/bioconda/muse}, v1.0.rc in 2-19-03

Current: GitHub, pre-build binary, v1.0rc

License: GPL 2

C++ 63.0\% C 36.8\% Makefile 0.2\%

Fan, Y., Xi, L., Hughes, D. S. T., Zhang, J., Zhang, J., Futreal, P. A., … Wang, W. (2016). MuSE: accounting for tumor heterogeneity using a sample-specific error model improves sensitivity and specificity in mutation calling from sequencing data. Genome Biology, 17(1), 178. \url{https://doi.org/10.1186/s13059-016-1029-6}

Updated: 2020-06-10

\section{MuTect1 (deprecated)}

Official: \url{https://software.broadinstitute.org/cancer/cga/mutect}

GitHub: \url{https://github.com/broadinstitute/mutect}, v1.1.6 in 2013-08-30

QBRC: \url{https://github.com/tianshilu/QBRC-Somatic-Pipeline/blob/master/somatic_script/mutect-1.1.7.jar}, v3.1-0-g72492bb (v1.1.7 in filename, v3.1-0-g72492bb in \verb|--version| command) in unknown

Current: QBRC, pre-build binary, v3.1-0-g72492bb

Java 96.3\% Perl 2.3\% Scala 1.4\%

Comment: Java 1.7.

Cibulskis, K., Lawrence, M. S., Carter, S. L., Sivachenko, A., Jaffe, D., Sougnez, C., … Getz, G. (2013). Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples. Nature Biotechnology, 31(3), 213–219. \url{https://doi.org/10.1038/nbt.2514}

Updated: 2020-06-10

\section{Shimmer}

GitHub: \url{https://github.com/nhansen/Shimmer}, v0.2 in 2018-03-07

Current: GitHub, pre-build binary, unknown

License: Public Domain -- United States Government Work

Perl 68.1\% C 11.4\% Python 11.0\% R 7.9\% Other 1.0\% Makefile 0.5\% Shell 0.1\%

Hansen, N. F., Gartner, J. J., Mei, L., Samuels, Y., \& Mullikin, J. C. (2013). Shimmer: Detection of genetic alterations in tumors using next-generation sequence data. Bioinformatics, 29(12), 1498–1503. \url{https://doi.org/10.1093/bioinformatics/btt183}

Updated: 2020-06-10

\section{Strelka2}

Conda: \url{https://anaconda.org/bioconda/strelka}, v2.9.10 in 2018-12

GitHub: \url{https://github.com/Illumina/strelka}, v2.9.10 in 2018-11-07

Current: GitHub, self-build, 2.9.10

License: GPL 3

C++ 81.3\% Python 12.8\% CMake 4.2\% Shell 1.1\% Dockerfile 0.3\% Jupyter Notebook 0.2\% C 0.1\%

Kim, S., Scheffler, K., Halpern, A. L., Bekritsky, M. A., Noh, E., Källberg, M., … Saunders, C. T. (2018). Strelka2: fast and accurate calling of germline and somatic variants. Nature Methods, 15(8), 591–594. \url{https://doi.org/10.1038/s41592-018-0051-x}

Updated: 2020-06-10
\section{VarScan1 (deprecated)}

Official: \url{http://varscan.sourceforge.net/}, v2.3.9 in 2015-06-03, deprecated

Current: N/A

Java

Koboldt, D. C., Chen, K., Wylie, T., Larson, D. E., McLellan, M. D., Mardis, E. R., Weinstock, G. M., Wilson, R. K. and Ding, L. (2009) VarScan: variant detection in massively parallel sequencing of individual and pooled samples., Bioinformatics (Oxford, England), 25(17), pp. 2283–5. doi: 10.1093/bioinformatics/btp373.

Updated: 2020-06-10

\section{VarScan2}

Official: \url{http://dkoboldt.github.io/varscan/}

GitHub: \url{https://github.com/dkoboldt/varscan}, v2.4.2 (release) in 2016-05-27, v2.4.4 (file) in 2019-07

Conda: \url{https://anaconda.org/bioconda/varscan}, v2.4.4 in 2019-10-26

Current: GitHub, pre-build binary, v2.4.2

License: The Non-Profit Open Software License version 3.0 (NPOSL-3.0)

Java

Koboldt, D. C., Zhang, Q., Larson, D. E., Shen, D., McLellan, M. D., Lin, L., Miller, C. A., Mardis, E. R., Ding, L. and Wilson, R. K. (2012) VarScan 2: Somatic mutation and copy number alteration discovery in cancer by exome sequencing, Genome Research. Cold Spring Harbor Laboratory Press, 22(3), pp. 568–576. doi: 10.1101/gr.129684.111.

Updated: 2020-06-10
\part{Variant Caller: CNV, Translocation and other types of SV}
\section{BreakDancer}

Official: \url{http://breakdancer.sourceforge.net/}

Conda: \url{https://anaconda.org/bioconda/breakdancer}, v1.4.5 in 2018-04

SorceForge: \url{https://sourceforge.net/projects/breakdancer}, v1.1.2\_2013\_03\_08 in 2013-03-06

License: GPL v3

Current: Conda, v1.4.5

Perl, C++

Chen, K., Wallis, J. W., McLellan, M. D., Larson, D. E., Kalicki, J. M., Pohl, C. S., McGrath, S. D., Wendl, M. C., Zhang, Q., Locke, D. P., Shi, X., Fulton, R. S., Ley, T. J., Wilson, R. K., Ding, L., \& Mardis, E. R. (2009). BreakDancer: An algorithm for high-resolution mapping of genomic structural variation. Nature Methods, 6(9), 677–681. \url{https://doi.org/10.1038/nmeth.1363}

Updated: 2020-06-25

\section{digit}

GitHub: \url{https://github.com/richard-meier/digit-trl/}, unknown version in 2016-12-06

Current: Github, pr-ebuild biniarirs, 2016-12-06

License: GPL v3

Java 100\%

Meier, R., Graw, S., Beyerlein, P., Koestler, D., Molina, J. R., \& Chien, J. (2017). digit-a tool for detection and identification of genomic interchromosomal  translocations. Nucleic Acids Research, 45(9), e72. \url{https://doi.org/10.1093/nar/gkx010}

Updated: 2020-06-25

\section{LUMPY}

GitHub: \url{https://github.com/arq5x/lumpy-sv}, v3.0 in 2019-02-21

Conda: \url{https://anaconda.org/bioconda/lumpy-sv}, v3.0 in 2019-11-28

Current: N/A

License: MIT

Comment: Use Python 2.7

C 76.8\% C++ 20.9\% Python 1.1\% Shell 0.7\% Makefile 0.3\% CMake 0.2\%

Layer, R. M., Chiang, C., Quinlan, A. R., \& Hall, I. M. (2014). LUMPY: A probabilistic framework for structural variant discovery. Genome Biology, 15(6), R84. \url{https://doi.org/10.1186/gb-2014-15-6-r84}

Updated: 2020-06-25

\section{Meerkat}

Official: \url{http://compbio.med.harvard.edu/Meerkat/}, v0.189

Current: N/A

Yang, L., Luquette, L. J., Gehlenborg, N., Xi, R., Haseley, P. S., Hsieh, C. H., Zhang, C., Ren, X., Protopopov, A., Chin, L., Kucherlapati, R., Lee, C., \& Park, P. J. (2013). Diverse mechanisms of somatic structural variations in human cancer genomes. Cell, 153(4), 919–929. \url{https://doi.org/10.1016/j.cell.2013.04.010}

Updated: 2020-05-26

\section{MetaSV}

GitHub: \url{https://github.com/bioinform/metasv}, v0.5.4 in 2017-01-18

Official: \url{bioinform.github.io/metasv/}

Conda: \url{https://anaconda.org/bioconda/metasv}, v0.5.4 in 2018-06

Current: N/A

Comment: Require Python 2.7-2.8

License: BSD-2-Clause License

Python 94.8\% HTML 5.1\% Shell 0.1\%

Mohiyuddin, M., Mu, J. C., Li, J., Bani Asadi, N., Gerstein, M. B., Abyzov, A., Wong, W. H., \& Lam, H. Y. K. (2015). MetaSV: an accurate and integrative structural-variant caller for next generation sequencing. Bioinformatics, 31(16), 2741–2744. \url{https://doi.org/10.1093/bioinformatics/btv204}

Updated: 2020-06-25

\section{SoftSV}

SourceForge: \url{https://sourceforge.net/projects/softsv/}, v1.4.2 in 2015-09-09

Current: N/A

License: GPL V3

C++

C. Bartenhagen, M. Dugas, Robust and exact structural variation detection with paired-end and soft-clipped alignments: SoftSV compared with eight algorithms, Brief Bioinform. (2015), \url{http://dx.doi.org/10.1093/bib/bbv028}

\section{SVMerge}

Official: \url{http://svmerge.sourceforge.net/}

Sourceforge: \url{https://sourceforge.net/projects/svmerge}, v1,2r37 in 2012-08-09

Conda: \url{https://anaconda.org/hcc/svmerge}, v1,2r37 in 2017-05

Current: Sourceforge, pre-built biniaries, v1,2r37 in 2012-08-09

License: GPL v3

Perl

Wong, K., Keane, T. M., Stalker, J., \& Adams, D. J. (2010). Enhanced structural variant and breakpoint detection using SVMerge by integration of multiple detection methods and local assembly. Genome Biology, 11(12), R128. \url{https://doi.org/10.1186/gb-2010-11-12-r128}

Updated: 2020-05-26

\part{VCF/BCF Processor}

\section{BCFTools}

Official: \url{http://www.htslib.org/}

Official: \url{http://samtools.github.io/bcftools/}

GitHub: \url{https://github.com/samtools/bcftools}, v1.10.2 in 2019-12-21

Conda: \url{https://anaconda.org/bioconda/bcftools}, v1.10.2 in 2019-12-22

Current: GitHub, self-build, 1.10.2 with htslib 1.10.2

License: MIT/Expat license or GPL 3

C 87.6\% Perl 9.5\% M4 0.9\% Python 0.7\% Makefile 0.6\% Shell 0.6\% Other 0.1\%

Citation same as Samtools.

Updated: 2020-06-10

\section{CrossMap}

SourceForge: 

Conda: \url{https://anaconda.org/bioconda/crossmap}, v0.33 in 2020-03-26

Official: \url{http://crossmap.sourceforge.net/}, v0.3.8 in 10/09/2019

Current: Conda, v0.33

License: GPL v2

Zhao, H., Sun, Z., Wang, J., Huang, H., Kocher, J.-P., \& Wang, L. (2014). CrossMap: a versatile tool for coordinate conversion between genome assemblies. Bioinformatics (Oxford, England), 30(7), 1006–1007. \url{https://doi.org/10.1093/bioinformatics/btt730}

Updated: 2020-06-15

\section{PLINK}

Official: \url{http://www.cog-genomics.org/plink/2.0/}, v2.00a3LM in 2020-01-15

GitHub: \url{https://github.com/chrchang/plink-ng}

Current: GitHub, pre-built binaries, v2.00a3LM in 2020-01-15

License: GPL v3

C 50.6\% C++ 47.5\% Python 1.1\% TeX 0.3\% Makefile 0.2\% Shell 0.1\% Other 0.2\% 

Chang, C. C., Chow, C. C., Tellier, L. C. A. M., Vattikuti, S., Purcell, S. M., \& Lee, J. J. (2015). Second-generation PLINK: rising to the challenge of larger and richer datasets. GigaScience, 4(1). \url{https://doi.org/10.1186/s13742-015-0047-8}

Hill, A., Loh, P.-R., Bharadwaj, R. B., Pons, P., Shang, J., Guinan, E., Lakhani, K., Kilty, I., \& Jelinsky, S. A. (2017). Stepwise Distributed Open Innovation Contests for Software Development: Acceleration of Genome-Wide Association Analysis. GigaScience, 6(5). \url{https://doi.org/10.1093/gigascience/gix009}

Manichaikul, A., Mychaleckyj, J. C., Rich, S. S., Daly, K., Sale, M., \& Chen, W.-M. (2010). Robust relationship inference in genome-wide association studies. Bioinformatics, 26(22), 2867–2873. \url{https://doi.org/10.1093/bioinformatics/btq559}

Galinsky, K. J., Bhatia, G., Loh, P. R., Georgiev, S., Mukherjee, S., Patterson, N. J., \& Price, A. L. (2016). Fast Principal-Component Analysis Reveals Convergent Evolution of ADH1B in Europe and East Asia. American Journal of Human Genetics, 98(3), 456–472. \url{https://doi.org/10.1016/j.ajhg.2015.12.022}

Yang, J., Lee, S. H., Goddard, M. E., \& Visscher, P. M. (2011). GCTA: A tool for genome-wide complex trait analysis. American Journal of Human Genetics, 88(1), 76–82. {https://doi.org/10.1016/j.ajhg.2010.11.011}

Gaunt, T. R., Rodríguez, S., \& Day, I. N. M. (2007). Cubic exact solutions for the estimation of pairwise haplotype frequencies: Implications for linkage disequilibrium analyses and a web tool “CubeX.” BMC Bioinformatics, 8(1), 428. \url{https://doi.org/10.1186/1471-2105-8-428}

Graffelman, J., \& Moreno, V. (2013). The mid p-value in exact tests for Hardy-Weinberg equilibrium. Statistical Applications in Genetics and Molecular Biology, 12(4), 433–448. \url{https://doi.org/10.1515/sagmb-2012-0039}

Graffelman, J., \& Weir, B. S. (2016). Testing for Hardy-Weinberg equilibrium at biallelic genetic markers on the X chromosome. Heredity, 116(6), 558–568. \url{https://doi.org/10.1038/hdy.2016.20}

Wigginton, J. E., Cutler, D. J., \& Abecasis, G. R. (2005). A note on exact tests of Hardy-Weinberg equilibrium. American Journal of Human Genetics, 76(5), 887–893. \url{https://doi.org/10.1086/429864}

Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A. R., Bender, D., Maller, J., Sklar, P., De Bakker, P. I. W., Daly, M. J., \& Sham, P. C. (2007). PLINK: A tool set for whole-genome association and population-based linkage analyses. American Journal of Human Genetics, 81(3), 559–575. \url{https://doi.org/10.1086/519795}

Updated:2020-06-16

\section{VCFTools}

SourceForge: \url{http://vcftools.sourceforge.net/}, v0.1.13 in 2015-08-03, branch deprecated

SVN: \url{http://svn.code.sf.net/p/vcftools/code/trunk/}

Official: \url{http://vcftools.github.io/}

GitHub: \url{https://github.com/vcftools/vcftools}, v0.1.16 in 2018-08-03

Current: GitHub, self-build, v0.1.17

Conda: \url{https://anaconda.org/bioconda/vcftools}, v0.1.16 in 2019-05

License: LGPL 3

C++ 44.8\% Perl 44.5\% C 6.6\% Roff 3.0\% Shell 0.8\% M4 0.2\% Makefile 0.1\%

Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., … Durbin, R. (2011). The variant call format and VCFtools. Bioinformatics, 27(15), 2156–2158. \url{https://doi.org/10.1093/bioinformatics/btr330}

Updated: 2020-06-10

\part{VCF Annotation \& Visualization}
\section{ANNOVar}

Official: \url{http://www.openbioinformatics.org/annovar/annovar_download.html}, v2014Dec12

Official: \url{https://doc-openbio.readthedocs.io/projects/annovar/en/latest/user-guide/download/}, v2019Oct24

Current: Official, pre-build binary, \$Date: 2019-10-24 00:05:27 -0400 (Thu, 24 Oct 2019) \$

Wang, K., Li, M., \& Hakonarson, H. (2010). ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Research, 38(16), e164–e164. \url{https://doi.org/10.1093/nar/gkq603}


Updated: 2020-06-10

\section{myVCF}

Official: \url{https://apietrelli.github.io/myVCF/}

ReadTheDocs: \url{https://myvcf.readthedocs.io/en/latest/}

GitHub: \url{https://github.com/apietrelli/myVCF}, v1.0 in 2017-10-09

Current: N/A

License: CC-BY-NC 4.0 International

Python 93.7\% HTML 6.2\% Other 0.1\%

Pietrelli, A. and Valenti, L. (2017) myVCF: a desktop application for high-throughput mutations data management, Bioinformatics, 33(22), pp. 3676–3678. doi: 10.1093/bioinformatics/btx475.

Updated: 2020-06-10
\section{snpEff}

Official: \url{http://snpeff.sourceforge.net/}, v4.3t in 2017-11-24

GitHub: \url{https://github.com/pcingola/SnpEff}, v4.3t in 2017-11-24

Conda: \url{https://anaconda.org/bioconda/snpeff}, v4.5covid19 in 2020-05-07

Current: Official, pre-build binary, 4.3t (build 2017-11-24 10:18)

Comment: Requires Java 1.8 (Java8)

License: LGPL 3

Java 62.0\% HTML 31.5\% Perl 2.2\% Shell 1.7\% CSS 0.8\% Python 0.6\% Other 1.2\%

Cingolani, P., Platts, A., Wang, L. L., Coon, M., Nguyen, T., Wang, L., Land, S. J., Lu, X. and Ruden, D. M. (2012) A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3, Fly. Taylor and Francis Inc., 6(2), pp. 80–92. doi: 10.4161/fly.19695.

Updated: 2020-06-10

\section{snpSift}

Official: \url{http://snpeff.sourceforge.net/}, v4.3t in 2017-11-24

GitHub: \url{https://github.com/pcingola/SnpSift}, v4.3t in 2017-11-24

Conda: \url{https://anaconda.org/bioconda/snpsift}, v4.3.1t in 2020-04-21

Current: Official, pre-build binary, v4.3t (build 2017-11-24 10:18)

License: LGPL 3

Comment: Requires Java 1.8 (Java8)

Java 97.3\% G-code 2.3\% Shell 0.4\%

Cingolani, P., Patel, V. M., Coon, M., Nguyen, T., Land, S. J., Ruden, D. M. and Lu, X. (2012) Using Drosophila melanogaster as a model for genotoxic chemical mutational studies with a new program, SnpSift, Frontiers in Genetics, 3(MAR). doi: 10.3389/fgene.2012.00035.

Updated: 2020-06-10


\section{Variant Effect Predicator}

GitHub \url{https://github.com/Ensembl/ensembl-vep}

Official: \url{https://asia.ensembl.org/info/docs/tools/vep/index.html}, v100.2 in 2020-05-25

Conda: \url{https://anaconda.org/bioconda/ensembl-vep}, v100.2 in 2020-06-01

Current: N/A

License: Apache 2.0

Perl 97.6\% Other 2.4\%

McLaren, W., Gil, L., Hunt, S. E., Riat, H. S., Ritchie, G. R. S., Thormann, A., Flicek, P. and Cunningham, F. (2016) The Ensembl Variant Effect Predictor, Genome Biology. BioMed Central Ltd., 17(1), p. 122. doi: 10.1186/s13059-016-0974-4.

Updated: 2020-06-10

\section{VarSifter}

GitHub: \url{https://github.com/teerjk/VarSifter}, v1.9 in 2018-08-01

Official: \url{https://research.nhgri.nih.gov/software/VarSifter/}, branch deprecated

FTP: \url{ftp://ftp.nhgri.nih.gov/pub/software/VarSifter/}, v1.7 in 2015-01-08

Current: N/A

License: THE JUNG LICENSE (Another kind of BSD)

Java 91.1\% HTML 8.4\% Makefile 0.5\%

Teer, J. K., Green, E. D., Mullikin, J. C., \& Biesecker, L. G. (2012). VarSifter: Visualizing and analyzing exome-scale sequence variation data on a desktop computer. Bioinformatics, 28(4), 599–600. \url{https://doi.org/10.1093/bioinformatics/btr711}

Updated: 2020-06-10

\part{Pipelines}
\section{SpeedSeq}

GitHub: \url{https://github.com/hall-lab/speedseq}, v0.1.2 in 2017-01-10

Current: N/A

License: The MIT License (MIT)

C 75.5\% Roff 8.4\% Perl 8.0\% Shell 3.1\% Makefile 1.5\% C++ 1.0\% Other 2.5\%

Chiang, C., Layer, R. M., Faust, G. G., Lindberg, M. R., Rose, D. B., Garrison, E. P., Marth, G. T., Quinlan, A. R. and Hall, I. M. (2015) SpeedSeq: Ultra-fast personal genome analysis and interpretation, Nature Methods. Nature Publishing Group, 12(10), pp. 966–968. doi: 10.1038/nmeth.3505.

Updated: 2020-06-10

\part{R, IDE and R Packages}

\section{R}

Conda: \url{https://anaconda.org/r/r-base}, v3.6.1 in 2020-05-20

Official: \url{https://www.r-project.org/}, v4.0.2 in 2020-06-22

SVN: \url{https://svn.r-project.org/R-dev-web/trunk/index.html}

Current: Conda, v4.0.0

License: GPL 2

R Core Team (2020) R: A Language and Environment for Statistical Computing. Vienna, Austria. Available at: \url{https://www.r-project.org}.

Updated: 2020-06-10

\section{RStudio}

Official: \url{https://rstudio.com/}, v1.3 in 2020-05-27

Conda: \url{https://anaconda.org/r/rstudio}, v1.1.456 in 2020-03-10

GitHub: \url{https://github.com/rstudio/rstudio}, v1.3.959 in 2020-05-27

Current: N/A

License: Commercial or AGPL 3

Java 36.1\% C++ 31/3\% JavaScript 20.4\% R 3.3\% C 3.2\% TypeScript 2.6\% Other 3.1\% 

RStudio Team (2015). RStudio: Integrated Development for R. RStudio, Inc., Boston, MA URL http://www.rstudio.com/.

Updated: 2020-06-10

\part{Python and Python Packages}

\section{Python3}

Conda: \url{https://anaconda.org/conda-forge/python}, v3.8.3 in 2020-06-02

Official: \url{https://www.python.org/}, v3.8.3 in 2020-05-13

GitHub: \url{https://github.com/python/cpython}, v3.9.0b3 in 2020-06-10

Current: Conda, v3.7.6

License: PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2

Python 63.0\% C 30.1\% Objective-C 4.2\% C++ 1.3\% HTML 0.4\% M4 0.4\% Other 0.6\% 

\textbf{From General Python 3 FAQ: \url{https://docs.python.org/3/faq/general.html}}

Are there any published articles about Python that I can reference?

It's probably best to cite your favourite book about Python.

The very first article about Python was written in 1991 and is now quite outdated.

Rossum, G. van and Boer, J. de (1991) Interactively Testing Remote Servers Using the Python Programming Language, CWI Quarterly, pp. 283–303.

Updated: 2020-06-10

\part{Docker and Docker Packages}

\part{File Specifications}
\section{FASTQ}

Rhizobium, G. E. (2013) Complete Genome Sequence of the Sesbania Symbiont and Rice, Nucleic acids research. Oxford Academic, 1(1256879), pp. 13–14. doi: 10.1093/nar.

Updated: 2020-06-10


\section{SAM, BAM, CRAM, BCF, VCF, CSI, BAI}

GitHub: \url{https://github.com/samtools/hts-specs}

Official: \url{http://samtools.github.io/hts-specs/}

TeX 97.8\% HTML 1.3\% Makefile 0.3\% Shell 0.2\% CSS 0.2\% JavaScript 0.1\% Dockerfile 0.1\% 

Sequence Alignment/Map Format (SAM) v1 in 2020-04-30

CRAM format v2.1 in 2019-01-22, Apache license 2.0

CRAM format v3 in 2020-01-20, Apache license 2.0

BCF v1 in 2019-11-20, deprecated

BCF v2.1 in 2019-07-24

CSI v1 in 2019-11-20

Tabix index file format in 2018-07-06

Variant Call Format (VCF) v4.1 in 2020-03-03

Variant Call Format (VCF) v4.2 in 2020-03-03

Variant Call Format (VCF) v4.3 in 3030-05-27

GA4GH File Encryption Standard (crypt4gh) in 2019-10-21

Htsget retrieval API spec v1.2.0 in 2020-02-20

Refget API Specification v1.0.0 in 2020-05-09

Updated: 2020-06-10

\end{document}
