\documentclass[]{article}
\usepackage[backref,bookmarks=true,bookmarksopen=true,pdftitle=SoftInstall,pdfauthor=YU Zhejian]{hyperref}
\renewcommand{\sfdefault}{phv}
\renewcommand{\rmdefault}{phv}
\newcommand{\cbb}[1]{}
\newcommand{\cb}[3]{\par Cited by: {\color{blue}\Huge #1} times in Scopus, {\color{blue}\Huge #2} times in Bing Academic, {\color{blue}\Huge #3} times in Baidu Xueshu}
\usepackage{indentfirst}
\renewcommand\familydefault{\sfdefault}
\usepackage[inner=0.5in,outer=0.5in]{geometry}
\usepackage{titletoc}
\usepackage{xcolor}
\usepackage{graphicx}
\titlecontents{part}[2em]{\addvspace{4pt}\large}{\contentslabel[\thecontentslabel]{4em}}{}{\ \titlerule*[.5pc]{.}\ \thecontentspage}[]
\titlecontents{section}[1em]{\addvspace{2pt}}{\bfseries\contentslabel[\thecontentslabel]{4em}}{}{\ \titlerule*[.5pc]{.}\ \thecontentspage}[]

\begin{document}
\title{\fontsize{40}{40}\selectfont\scshape Fantastic Software\\{\Huge and}\\Where to Find Them}
\author{YU Zhejian}
\maketitle
\setcounter{section}{-1}
\begin{center}
\includegraphics[width=0.5\linewidth]{YuZJLab_BG.png}
\end{center}
\section{Introduction}

This book involves the software used in the common bioinformatics pipelines and the way to install them.

The term ``\textbf{deprecated}'' means that the entire software was deprecated by its developer, while ``\textbf{branch deprecated}'' means that although this branch was deprecated, the software is still being developed somewhere else. Unless there is \textbf{EXPLICIT} information, all branches \& software are considered not deprecated.

The term ``pre-build binary'' means 1. Compiled binary for C, C++ or D, etc. or 2. Compiled documentations for \LaTeX or asciidoc, etc. or 3. Scripts for scripting languages like Perl, Shell or Python, etc. Those which is not ``pre-build binary'' are compiled locally by ourselves. ``N/A'' means we have not, or failed to install it.

As for tag ``Cited by'', ``0'' means this article is cited 0 times in Scopus (\url{https://www.scopus.com/}) and ``N/A'' means this article do not present in Scopus. Same for \url{https://xueshu.baidu.com/} and \url{https://academic.microsoft.com/home}.

\tableofcontents

\part{2nd Gen. Read Data Pre-Processing and 2nd Gen. Librries}

\section{BamTools}

Sourceforge: \url{https://sourceforge.net/projects/bamtools/}, deprecated

GitHub: \url{https://github.com/pezmaster31/bamtools}, v2.5.1, 2019-12-24

Current: GitHub, self-built binary, v2.5.1

License: MIT

C++ 96.4\% Python 1.8\% CMake 1.3\% Other 0.5\%

Barnett, D. W., Garrison, E. K., Quinlan, A. R., Strömberg, M. P., \& Marth, G. T. (2011). BamTools: a C++ API and toolkit for analyzing and managing BAM files. Bioinformatics, 27(12), 1691–1692. \url{https://doi.org/10.1093/bioinformatics/btr1744}\cb{267}{426}{250}

Updated: 2020-07-21

\section{BLAT}

Official: \url{http://www.soe.ucsc.edu/~kent}, v36 in 2014-11-05

Current: Official, self-built binary, v36

Updated: 2020-07-21

\section{CAP3}

Official: \url{http://seq.cs.iastate.edu/cap3.html}, unknown

Current: Official, pre-built binary

FORTRAN, perl

Updated: 2020-07-21

\section{gfatools}

GitHub: \url{https://github.com/lh3/gfatools}, v0.4 (r165) in 2019-12-23

C 96.8\% Objective-C 1.9\% Other 1.3\%

Current: GitHub, self-built binary, 0.4-r179-dirty with gfa.h: 0.4-r186-dirty

Updated: 2020-07-21

\section{htsjdk}

GitHub: \url{https://github.com/samtools/htsjdk}, v2.23.0 in 2020-07-08

Official: \url{http://samtools.github.io/htsjdk/}

Current: N/A

Java 99.8\% Other 0.2\%

Updated: 2020-07-21

\section{htslib}

GitHub: \url{https://github.com/samtools/htslib} with MIT/Expat license, Modified 3-Clause BSD license, v1.10.2 in 2019-12-21 

Conda: \url{https://anaconda.org/bioconda/htslib} with MIT License, v1.10.2 in 2019-12-23

Official: \url{http://www.htslib.org/}

Current: GitHub, self-build,  1.10.2-23-g6b72368

C 89.3\% C++ 3.7\% Perl 2.7\% Makefile 1.4\% Roff 1.2\% M4 1.0\% Other 0.7\%

Updated: 2020-07-21

\section{SeqAn2}

Official: \url{https://www.seqan.de/}

GitHub: \url{https://github.com/seqan/seqan}, v2.4.0 in 2018-02-03

Current: GitHub, master

License: The 3-Clause BSD License

C++ 63.9\% PHP 22.6\% Python 3.4\% HTML 2.7\% JavaScript 2.3\% C 1.7\% Other 3.4\%

Döring, A., Weese, D., Rausch, T., \& Reinert, K. (2008). SeqAn an efficient, generic C++ library for sequence analysis. BMC Bioinformatics, 9. \url{https://doi.org/10.1186/1471-2105-9-11} \cb{173}{342}{281}

Updated: 2020-07-21

\section{SeqAn3}

Official: \url{https://www.seqan.de/}

GitHub: \url{https://github.com/seqan/seqan3}, v3.0.1 in 2020-01-22

Current: GitHub, master

License: The 3-Clause BSD License

C++ 98.9\% CMake 1.1\%

Updated: 2020-07-21

\section{SRA-Toolkit}

Conda: \url{https://anaconda.org/bioconda/sra-tools}, v2.10.8 in 2020-07-04

Official: \url{https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software}

GitHub: \url{https://github.com/ncbi/sra-tools/}, v2.10.8 in 2020-06-29

Current: GitHub, pre-build binary, 2.10.4

License: PUBLIC DOMAIN NOTICE

C 70.3\% C++ 19.3\% Makefile 4.0 Shell 2.3\% Perl 1.9\% Python 1.0\% Other 1.2\%

Updated: 2020-07-21

\section{Tabix (deprecated)}

Conda: \url{https://anaconda.org/bioconda/tabix}, v0.2.6 in 2018-10

GitHub: \url{https://github.com/samtools/tabix}

SourceForge: \url{https://sourceforge.net/projects/samtools}, branch deprecated

Current: N/A

License: BSD

C 83.8\% Java 6.8\% Python 4.6\% TeX 2.8\% Perl 2.0\%

Updated: 2020-07-21

\part{2nd Gen. Read Quality Controller}

\section{CutAdapt}

Conda: \url{https://anaconda.org/bioconda/cutadapt}, v2.10 in 2020-07-13

GitHub: \url{https://github.com/marcelm/cutadapt}, v2.10 in 2020-04-22

Official: \url{https://cutadapt.readthedocs.io/en/stable/}

CutAdapt: Conda, 2.10

License: MIT

Python 99.6\% Shell 0.4\%

Martin, M. (2011). Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet.Journal, 17(1), 10. \url{https://doi.org/10.14806/ej.17.1.200} \cb{N/A}{11649}{2032}

Updated: 2020-07-21

\section{FastQC}

Conda: \url{https://anaconda.org/bioconda/fastqc} with GPL 3, v0.11.9 in 2020-02-23

Official: \url{http://www.bioinformatics.babraham.ac.uk/projects/fastqc/} with GPL 3

GitHub: \url{https://github.com/s-andrews/FastQC} with GPL 2, v0.11.9 in 2020-01-08

Current: GitHub, pre-build binary, 0.11.9

Java 95.5\% HTML 3.1\% Perl 1.2\% Other 0.2\%

Updated: 2020-07-21

\section{QualiMap}

Official: \url{http://qualimap.bioinfo.cipf.es/}

BitBucket: \url{https://bitbucket.org/kokonech/qualimap/}, build 11-11-19 in 2019-11-11

Current: N/A

License: GPL 2

Java

García-Alcalde, F., Okonechnikov, K., Carbonell, J., Cruz, L. M., Götz, S., Tarazona, S., Dopazo, J., Meyer, T. F. and Conesa, A. (2012) Qualimap: evaluating next-generation sequencing alignment data, Bioinformatics, 28(20), pp. 2678–2679. doi: 10.1093/bioinformatics/bts503.\cb{275}{431}{326}

Okonechnikov, K., Conesa, A. and García-Alcalde, F. (2015) Qualimap 2: advanced multi-sample quality control for high-throughput sequencing data, Bioinformatics, 32(2), pp. 292–294. doi: 10.1093/bioinformatics/btv566.\cb{271}{427}{145}

Updated: 2020-07-21

\section{Trimmomatic}

Official: \url{http://www.usadellab.org/cms/index.php?page=trimmomatic}

GitHub: \url{https://github.com/timflutre/trimmomatic}, unknown

Conda: \url{https://anaconda.org/bioconda/trimmomatic}, v0.39 in 2019-11-21

Current: GitHub, pre-build binary, v0.39

License: GPL 3

Java 99.2\% Makefile 0.8\%

Bolger, A. M., Lohse, M., \& Usadel, B. (2014). Trimmomatic: A flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114–2120.\url{https://doi.org/10.1093/bioinformatics/btu170} \cb{12021}{16246}{7227}

Updated: 2020-07-21

\part{3rd Gen. Read Quality Controller}

\section{REQUEST}

GitHub: \url{https://github.com/bioinfomaticsCSU/REQUEST}, unknown

Current: GitHub, master

License: Unknown

Zhang, W., Huang, N., Zheng, J., Liao, X., Wang, J., \& Li, H. D. (2019). A Sequence-Based Novel Approach for Quality Evaluation of Third-Generation Sequencing Reads. Genes, 10(1), 44. \url{https://doi.org/10.3390/genes10010044}\cb{0}{0}{0}

Updated: 2020-07-21

\part{2nd Gen. Aligner}
\section{Bowtie}

Official: \url{http://bowtie-bio.sourceforge.net/index.shtml}, v1.2.3, 2019-07-05

GitHub: \url{https://github.com/BenLangmead/bowtie}, v1.2.3 in 2019-07-06

Conda: \url{https://anaconda.org/bioconda/bowtie}, v1.2.3 in 2020-05-09

Current: GitHub, pre-build binary

License: Artistic License 2.0

C++ 91.1\% Perl 6.2\% Shell 0.9\% C 0.7\% Python 0.6\% Makefile 0.4\% Objective-C 0.1\%

Langmead, B., Trapnell, C., Pop, M., \& Salzberg, S. L. (2009). Ultrafast and memory-efficient alignment of short DNA sequences to the human genome. Genome Biology, 10(3), R25. \url{https://doi.org/10.1186/gb-2009-10-3-r25} \cb{11653}{16910}{13100}

Updated: 2020-07-21

\section{Bowtie2}

Conda: \url{https://anaconda.org/bioconda/bowtie2}, v2.4.1 in 2020-06-07

Official: \url{http://bowtie-bio.sourceforge.net/bowtie2/index.shtml}, v2.4.1 in 2020-02-28

GitHub: \url{https://github.com/BenLangmead/bowtie2}, v2.4.1 in 2020-02-29

Current: GitHub, pre-build binary

License: GPL 3

C++ 80.7\% Perl 14.5\% Python 1.5\% C 1.3\% Shell 1.1\% Makefile 0.6\% CMake 0.3\%

Langmead, B., Salzberg, S. (2012) Fast gapped-read alignment with Bowtie 2. Nature Methods, 9(4), 357–359. \url{https://doi.org/10.1038/nmeth.1923}\cb{14114}{20775}{9304}

Updated: 2020-07-21

\section{BWA}

Conda: \url{https://anaconda.org/bioconda/bwa}, v0.7.17 in 2020-07-10

GitHub: \url{https://github.com/lh3/bwa}, v0.7.17-r1188 in 2017-10-24

Official: \url{http://bio-bwa.sourceforge.net/}

SourceForge: \url{https://sourceforge.net/projects/bio-bwa/}, v0.7.17 in 2017

Current: GitHub, self-build, v0.7.17-r1188

License: GPL 3

C 85.7\% JavaScript 5.8\% Roff 4.3\% C++ 1.8\% Perl 1.3\% Shell 0.6\% Makefile 0.5\%

\textbf{BWA-backtrack}

Li, H., \& Durbin, R. (2009). Fast and accurate short read alignment with Burrows-Wheeler transform. Bioinformatics, 25(14), 1754–1760. \url{https://doi.org/10.1093/bioinformatics/btp324}\cb{16923}{24572}{18429}

\textbf{BWA-SW}

Li, H., \& Durbin, R. (2010). Fast and accurate long-read alignment with Burrows-Wheeler transform. Bioinformatics, 26(5), 589–595. \url{https://doi.org/10.1093/bioinformatics/btp698}\cb{4464}{N/A}{0}

\textbf{BWA-MEM}

Li, H. (2013) Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. Available at: \url{http://arxiv.org/abs/1303.3997} (Accessed: 2020-07-21).\cb{N/A}{4118}{1490}

Updated: 2020-07-21

\section{HISAT (deprecated)}

Official: \url{https://ccb.jhu.edu/software/hisat/index.shtml}, v0.1.6-beta in 2015-04-17

GitHub: \url{https://github.com/infphilo/hisat}

Current: N/A

License: GPL v3

C++ 67.1\% C 22.9\% Perl 8.9\% Other 1.1\%

Kim, D., Langmead, B. \& Salzberg, S. HISAT: a fast spliced aligner with low memory requirements. Nat Methods 12, 357–360 (2015). \url{https://doi.org/10.1038/nmeth.3317}\cb{3191}{4568}{1088}

Updated: 2020-07-21

\section{HiSat2}

Conda: \url{https://anaconda.org/bioconda/hisat2}, v2.2.0 in 2020-03-02

Official: \url{https://daehwankimlab.github.io/hisat2/}, v2.2.0 in 2020-02-07

GitHub: \url{https://github.com/DaehwanKimLab/hisat2}

Current: Official, pre-build binary, 2.2.0

License: GPL 3

C++ 57.8\% C 16.9\% Python 15.9\% Perl 6.2\% Shell 1.3\% HTML 1.3\% Other 0.6\%

Updated: 2020-07-21

\section{SubRead}

Conda: \url{https://anaconda.org/bioconda/subread}, v2.0.1 in 2020-06-05

SourceForge: \url{https://sourceforge.net/projects/subread}, v2.0.1 in 2020-05-13

Official: \url{http://subread.sourceforge.net/}

Current: Official, pre-build binary, v2.0.0

License: GPL 3

Liao, Y., Smyth, G. K., \& Shi, W. (2013). The Subread aligner: Fast, accurate and scalable read mapping by seed-and-vote. Nucleic Acids Research, 41(10). \url{https://doi.org/10.1093/nar/gkt214}\cb{867}{1321}{450}

Liao, Y., Smyth, G. K. and Shi, W. (2019) The R package Rsubread is easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads, Nucleic acids research. NLM (Medline), 47(8), p. e47. doi: 10.1093/nar/gkz114.\cb{73}{142}{0}

Liao, Y., Smyth, G. K. and Shi, W. (2014) FeatureCounts: An efficient general purpose program for assigning sequence reads to genomic features, Bioinformatics. Oxford University Press, 30(7), pp. 923–930. doi: 10.1093/bioinformatics/btt656.\cb{3136}{4900}{1709}

Updated: 2020-07-21

\section{TopHat2 (deprecated)}

Official: \url{http://ccb.jhu.edu/software/tophat/index.shtml}, v2.1.1 in 2020-02-23

GitHub: \url{https://github.com/DaehwanKimLab/tophat}, v2.1.2 in 2018-05-24

Conda: \url{https://anaconda.org/bioconda/tophat}, v2.1.1 in 2019-05

Current: N/A

License: BSL 1.0

C++ 84.8\% C 10.0\% Python 3.8\% Perl 0.6\% Makefile 0.3\% M4 0.2\% Other 0.3\%

Trapnell, C., Pachter, L. and Salzberg, S. L. (2009) TopHat: discovering splice junctions with RNA-Seq., Bioinformatics (Oxford, England), 25(9), pp. 1105–1111. doi: 10.1093/bioinformatics/btp120.\cb{7361}{10513}{8098}

Langmead, B., Trapnell, C., Pop, M. and Salzberg, S. L. (2009) Ultrafast and memory-efficient alignment of short DNA sequences to the human genome, Genome Biology. BioMed Central, 10(3), p. R25. doi: 10.1186/gb-2009-10-3-r25.\cb{11653}{16910}{13094}

Kim, D., Pertea, G., Trapnell, C., Pimentel, H., Kelley, R. and Salzberg, S. L. (2013) TopHat2: Accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions, Genome Biology. BioMed Central, 14(4), p. R36. doi: 10.1186/gb-2013-14-4-r36.\cb{6250}{8936}{5655}

Kim, D. and Salzberg, S. L. (2011) TopHat-Fusion: An algorithm for discovery of novel fusion transcripts, Genome Biology. BioMed Central, 12(8), p. R72. doi: 10.1186/gb-2011-12-8-r72.\cb{432}{625}{444}

Updated: 2020-07-21

\part{3rd Gen. Aligner}

\section{ngmlr}

GitHub: \url{https://github.com/philres/ngmlr}, v0.2.7 in 2018-07-25

Conda: \url{https://anaconda.org/bioconda/ngmlr}, v0.2.7 in 2019-03

Current: GitHub, self-built biniariy, v0.2.8

License: MIT

C++ 82.8\% C 10.9\% Python 3.4\% Makefile 1.7\% Shell 0.7\% CMake 0.4\% Dockerfile 0.1\% 

Sedlazeck, F.J., Rescheneder, P., Smolka, M. et al. Accurate detection of complex structural variations using single-molecule sequencing. Nat Methods 15, 461–468 (2018). \url{https://doi.org/10.1038/s41592-018-0001-7}\cb{155}{302}{50}

Updated: 2020-07-21

\section{minimap2}

Official: \url{https://lh3.github.io/minimap2/}

GitHub: \url{https://github.com/lh3/minimap2}, v2.17(r941) in 2019-05-05

Conda: \url{https://anaconda.org/bioconda/minimap2}, v2.17 in 2020-06-20

Current: GitHub, self-built binary, v2.17-r974-dirty

Comment: Use \verb|clang|.

License: MIT

C 50.7\% TeX 17.0\% JavaScript 12.9\% C++ 12.0\% Roff 2.9\% Python 2.4\% Other 2.1\%

Heng Li, Minimap2: pairwise alignment for nucleotide sequences, Bioinformatics, Volume 34, Issue 18, 15 September 2018, Pages 3094–3100, \url{https://doi.org/10.1093/bioinformatics/bty191}\cb{421}{1040}{75}

Updated: 2020-07-21

\section{pbmm2}

GitHUb: \url{https://github.com/PacificBiosciences/pbmm2}, v1.2.1 in 2020-02-18

Conda: \url{https://anaconda.org/bioconda/pbmm2}, v1.3.0 in 2020-05-21

Current: N/A

License: BSD-3-Clause-Clear License

C++ 39.9\% Perl 29.8\% Raku 23.2\% Python 4.0\% Meson 1.7\% Shell 1.4\%

Updated: 2020-07-21

\part{2nd Gen. SAM/BAM/CRAM File Processor}

\section{AlignStats}

GitHub: \url{https://github.com/jfarek/alignstats}, v0.9.1 in 2020-02-08

Conda: \url{https://anaconda.org/bioconda/alignstats}, v0.9.1 in 2020-02-08

Current: GitHub, self-built binary, v0.9.1

License: BSD-3-Clause License

C 97.4\% M4 2.0\% Makefile 0.6\%

Updated: 2020-07-21

\section{BamDSt}

GitHub: \url{https://github.com/shiquan/bamdst}

Current: GitHub, self-built binary, v1.0.9

License: MIT

C 95.5\% C++ 3.9\% Makefile 0.6\%

Updated: 2020-07-21

\section{BamStats (by biopet)}

GitHub: \url{https://github.com/biopet/bamstats}, v1.0.1 in 2018-10-25

Conda: \url{https://anaconda.org/bioconda/biopet-bamstats}, v1.0.1 in 2018-11

Current: N/A

License: MIT License

Scala 100.0\%

Updated: 2020-07-21

\section{BAMStats (by nestornotabilis)}

Official: \url{http://bamstats.sourceforge.net/}, v1.25 in 2011-08-18

Current: Official, Pre-built binary, v1.25

Updated: 2020-07-21

\section{BamUtil}

Official: \url{https://genome.sph.umich.edu/wiki/BamUtil}

GitHub: \url{https://github.com/statgen/bamUtil}, v1.0.14 in 2017-07-01

Conda: \url{https://anaconda.org/bioconda/bamutil}, v1.0.14 in 2018-05-03

Current: GitHub, self-built binary, v1.0.14

License: GPL v3

C++ 79.8\% Shell 19.5\% Other 0.7\%

Jun, G., Wing, M. K., Abecasis, G. R., \& Kang, H. M. (2015). An efficient and scalable analysis framework for variant extraction and refinement from population-scale DNA sequence data. Genome Research, 25(6), 918–925. \url{https://doi.org/10.1101/gr.176552.114}\cb{54}{99}{36}

Updated: 2020-07-21

\section{BedTools}

GitHub: \url{https://github.com/arq5x/bedtools2/}, v2.29.2 in 2019-12-28

ReadtheDocs: \url{http://bedtools.readthedocs.org}

Conda: \url{https://anaconda.org/bioconda/bedtools}, v2.29.2 in 2020-07-03

Current: GitHub, pre-built binary, v2.29.2

License: MIT

C 49.6\% C++ 35.5\% Shell 10.0\% Makefile 1.7\% HTML 1.5\% Roff 0.8\% Other 0.9\%

\textbf{Please cite the following article if you use BEDTools in your research:}

Quinlan AR and Hall IM, 2010. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 26, 6, pp. 841–842.  doi: 10.1093/bioinformatics/btq033 \cb{6630}{10339}{6779}

\textbf{Also, if you use pybedtools, please cite the following.}

Dale RK, Pedersen BS, and Quinlan AR. Pybedtools: a flexible Python library for manipulating genomic datasets and annotations. Bioinformatics (2011). doi:10.1093/bioinformatics/btr539\cb{117}{200}{146}

Updated: 2020-07-21

\section{Picard}

Official: \url{http://broadinstitute.github.io/picard/}

Conda: \url{https://anaconda.org/bioconda/picard}, v2.22.9 in 2020-06-03

GitHub: \url{https://github.com/broadinstitute/picard}, v2.23.3 in 2020-07-21

Current: GitHub, pre-build binary, v2.21.6-SNAPSHOT

License: MIT

Java 98.7\% R 0.4\% XSLT 0.4\% Shell 0.2\% HTML 0.2\% CSS 0.1\% 

Updated: 2020-07-21

\section{Sambamba}

Conda: \url{https://anaconda.org/bioconda/sambamba}, v0.7.1 in 2020-03-15 

Official: \url{http://www.sambamba.org/}

Related: \url{https://thebird.nl/blog/D_Dragon.html}

GitHub: \url{https://github.com/biod/sambamba}, v0.7.1 (20191128) in 2019-11-29

Current: GitHub, pre-build binary, 0.7.1

\begin{verbatim}
LDC 1.17.0 / DMD v2.087.1 / LLVM8.0.1 / bootstrap LDC - the LLVM D compiler
(1.17.0)
\end{verbatim}

License: GPL 2

D 82.0\% Shell 7.7\% Roff 6.0\% Python 1.6\% Ruby 1.1\% Meson 0.8\% Makefile 0.8\%

Tarasov, A., Vilella, A. J., Cuppen, E., Nijman, I. J., \& Prins, P. (2015). Sambamba: fast processing of NGS alignment formats. Bioinformatics, 31(12), 2032–2034. \url{https://doi.org/10.1093/bioinformatics/btv098}\cb{229}{381}{82}

Updated: 2020-07-21

\section{SAMBLASTER}

GitHub: \url{https://github.com/GregoryFaust/samblaster}, v0.1.26 in 2020-07-05

Conda: \url{https://anaconda.org/bioconda/samblaster}, v0.1.26 in 2020-06-16

Current: Github, self-build binary, v0.1.26

License: MIT

C++ 99.4\% Makefile 0.6\%

Faust, G. G., \& Hall, I. M. (2014). SAMBLASTER: fast duplicate marking and structural variant read extraction. Bioinformatics, 30(17), 2503–2505. \url{https://doi.org/10.1093/bioinformatics/btu314} \cb{161}{253}{111}

Updated: 2020-07-21

\section{Samtools}

Official: \url{http://www.htslib.org/}

SourceForge: \url{http://samtools.sourceforge.net/}, deprecated

Conda: \url{https://anaconda.org/bioconda/samtools}, v1.10 in 2020-04-16

GitHub: \url{https://github.com/samtools/samtools}, v1.10 in 2019-12-07

Current: GitHub, self-build, v1.10 with htslib v1.10

License: The MIT/Expat License

C 73.3\% Perl 18.8\% M4 2.7\% Lua 1.8\% Shell 1.4\% Makefile 1.0\% Other 1.0\%

\textbf{VCF Format:}

Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., … Durbin, R. (2011). The variant call format and VCFtools. Bioinformatics, 27(15), 2156–2158. \url{https://doi.org/10.1093/bioinformatics/btr330} \cb{3238}{5425}{2901}

\textbf{SAM \& BAM Format:}

Cock, P. J. A., Bonfield, J. K., Chevreux, B., \& Li, H. (2015). SAM/BAM format v1.5 extensions for de novo assemblies. BioRxiv, 020024. \url{https://doi.org/10.1101/020024} \cb{N/A}{4}{0}

Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., … Durbin, R. (2009). The Sequence Alignment/Map format and SAMtools. Bioinformatics, 25(16), 2078–2079. \url{https://doi.org/10.1093/bioinformatics/btp352} \cb{18682}{27828}{16119}

\textbf{CRAM Format:}

Fritz, M. H. Y., Leinonen, R., Cochrane, G., \& Birney, E. (2011). Efficient storage of high throughput DNA sequencing data using reference-based compression. Genome Research, 21(5), 734–740. \url{https://doi.org/10.1101/gr.114819.110}\cb{202}{331}{331}

\textbf{The original mpileup calling algorithm plus mathematical notes (mpileup/bcftools call -c)}

Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics, 27(21), 2987–2993. \url{https://doi.org/10.1093/bioinformatics/btr509}\cb{1525}{2418}{0}

LI, H. (2010). Mathematical Notes on SAMtools Algorithms.

\textbf{Mathematical notes for the updated multiallelic calling model (mpileup/bcftools call -m)}

Petr Danecek, Stephan Schiffels, R. D. (2016). Multiallelic calling model in bcftools (-m). Retrieved January 30, 2020, from \url{http://samtools.github.io/bcftools/call-m.pdf}\cb{N/A}{N/A}{N/A}

\textbf{Hidden Markov model for detecting runs of homozygosity (bcftools roh)}

Narasimhan, V., Danecek, P., Scally, A., Xue, Y., Tyler-Smith, C., \& Durbin, R. (2016). BCFtools/RoH: A hidden Markov model approach for detecting autozygosity from next-generation sequencing data. Bioinformatics, 32(11), 1749–1751. \url{https://doi.org/10.1093/bioinformatics/btw044} \cb{98}{162}{16}

\textbf{Copy number variation/aneuploidy calling from microarray data (bcftools cnv/bcftools polysomy)}

Danecek, P., McCarthy, S. A., Consortium, H. S., \& Durbin, R. (2016). A method for checking genomic integrity in cultured cell lines from snp genotyping data. PLoS ONE, 11(5). \url{https://doi.org/10.1371/journal.pone.0155014}\cb{9}{10}{8}

\textbf{Haplotype-aware calling of variant consequences (bcftools csq)}

Danecek, P., \& McCarthy, S. A. (2017). BCFtools/csq: Haplotype-aware variant consequences. Bioinformatics, 33(13), 2037–2039. \url{https://doi.org/10.1093/bioinformatics/btx100}\cb{31}{71}{3}

\textbf{Base alignment quality (BAQ) method improve SNP calling around INDELs}

Li, H. (2011). Improving SNP discovery by base alignment quality. Bioinformatics, 27(8), 1157–1158. \url{https://doi.org/10.1093/bioinformatics/btr076}\cb{129}{200}{217}

\textbf{Segregation based QC metric originally implemented in SGA}

Durbin, R. (2014). Segregation based metric for variant call QC. Retrieved from \url{http://samtools.github.io/bcftools/rd-SegBias.pdf}\cb{N/A}{N/A}{N/A}

Updated: 2020-07-21

\part{2nd Gen. Variant Caller: SNP and Indel}
\section{FreeBayes}

GitHub: \url{https://github.com/ekg/freebayes}, v1.3.2 in 2019-12-17

Conda: \url{https://anaconda.org/bioconda/freebayes}, v1.3.2 in 2020-05-09

Current: GitHub, pre-build binary, v1.3.1-dirty

License: MIT

C++ 95.7\% TeX 2.1\% Python 0.9\% Shell 0.5\% C 0.4\% Assembly 0.3\% Other 0.1\%

Garrison, E., \& Marth, G. (2012). Haplotype-based variant detection from short-read sequencing. Retrieved from \url{http://arxiv.org/abs/1207.3907}\cb{N/A}{1979}{1038}

Updated: 2020-07-21

\section{GATK3 (deprecated)}

Conda: \url{https://anaconda.org/bioconda/gatk}, v3.8 in 2019-10-17

Docker: \url{https://hub.docker.com/r/broadinstitute/gatk3/}, v3.8.1 in 2014

GoogleCloud: \url{https://console.cloud.google.com/storage/browser/gatk-software/package-archive/gatk/}, v3.8-1-0-gf15c1c3ef in 2019-12-09

Current: GoogleCloud, pre-build binary, 3.8-1-0-gf15c1c3ef

Updated: 2020-07-21

\section{GATK4}

Official: \url{https://software.broadinstitute.org/gatk/}

GitHub: \url{https://github.com/broadinstitute/gatk}, v4.1.8.1 in 2020-07-21

Conda: \url{https://anaconda.org/bioconda/gatk4}, v4.1.7.0 in 2020-05-23

Docker: \url{https://hub.docker.com/r/broadinstitute/gatk/}, v4.1.7.0 in 2020-04

Current: GitHub, pre-build binary

\begin{verbatim}
Using GATK jar gatk-package-4.1.4.1-local.jar
Running:
java -Dsamjdk.use_async_io_read_samtools=false
-Dsamjdk.use_async_io_write_samtools=true
-Dsamjdk.use_async_io_write_tribble=false
-Dsamjdk.compression_level=2 -jar 
gatk-package-4.1.4.1-local.jar --version
The Genome Analysis Toolkit (GATK) v4.1.4.1
HTSJDK Version: 2.21.0
Picard Version: 2.21.2
\end{verbatim}

License: BSD-3-Clause

Java 93.8\% Python 2.8\% wdl 1.9\% Shell 1.1\% R 0.3\% HTML 0.1\%

\textbf{The first GATK paper covers the computational philosophy underlying the GATK and is a good citation for the GATK in general.}

McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., … DePristo, M. A. (2010). The genome analysis toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data. Genome Research, 20(9), 1297–1303. \url{https://doi.org/10.1101/gr.107524.110}\cb{9332}{13558}{8060}

\textbf{The second GATK paper describes in more detail some of the key tools commonly used in the GATK for high-throughput sequencing data processing and variant discovery. The paper covers base quality score recalibration, indel realignment, SNP calling with UnifiedGenotyper, variant quality score recalibration and their application to deep whole genome, whole exome, and low-pass multi-sample calling. This is a good citation if you use the GATK for variant discovery.}

Depristo, M. A., Banks, E., Poplin, R., Garimella, K. V., Maguire, J. R., Hartl, C., … Daly, M. J. (2011). A framework for variation discovery and genotyping using next-generation DNA sequencing data. Nature Genetics, 43(5), 491–501. \url{https://doi.org/10.1038/ng.806}\cb{5158}{7709}{7561}

\textbf{The third GATK paper describes the Best Practices for Variant Discovery (version 2.x). It is intended mainly as a learning resource for first-time users and as a protocol reference. This is a good citation to include in a Materials and Methods section.}

Van der Auwera, G. A., Carneiro, M. O., Hartl, C., Poplin, R., del Angel, G., Levy-Moonshine, A., … DePristo, M. A. (2013). From fastQ data to high-confidence variant calls: The genome analysis toolkit best practices pipeline. Current Protocols in Bioinformatics, (SUPL.43). \url{https://doi.org/10.1002/0471250953.bi1110s43}\cb{1851}{2965}{1380}

\textbf{The fourth paper, technically just a manuscript deposited in bioRxiv -- but it counts! This is a good citation to include in a Materials and Methods section or in a Discussion if you're talking about the joint calling process.}

Poplin, R., Ruano-Rubio, V., DePristo, M. A., Fennell, T. J., Carneiro, M. O., Auwera, G. A. Van der, … Banks, E. (2017). Scaling accurate genetic variant discovery to tens of thousands of samples. BioRxiv, 201178. \url{https://doi.org/10.1101/201178}\cb{N/A}{178}{25}

Updated: 2020-07-21

\section{LoFreq}

SourceForge: \url{https://sourceforge.net/projects/lofreq/}, v2.1.2 in 2015-05-19

Conda: \url{https://anaconda.org/bioconda/lofreq}, v2.1.4 in 2020-05-09

Official: \url{http://csb5.github.io/lofreq/}

GitHub: \url{https://github.com/CSB5/lofreq}, v2.1.4 in 2020-01-05

Current: GitHub, pre-build binary

\begin{verbatim}
version: 2.1.4
commit: unknown
build-date: Feb 19 2020
\end{verbatim}

License: MIT License

C 74.2\% Python 17.8\% Shell 5.3\% M4 2.3\% Other 0.4\%

Wilm, A., Aw, P. P. K., Bertrand, D., Yeo, G. H. T., Ong, S. H., Wong, C. H., … Nagarajan, N. (2012). LoFreq: A sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets. Nucleic Acids Research, 40(22), 11189–11201. \url{https://doi.org/10.1093/nar/gks918}\cb{324}{486}{163}

Updated: 2020-07-21
\section{Manta}

Conda: \url{https://anaconda.org/bioconda/manta}, v1.6.0 in 2019-10-06

GitHub: \url{https://github.com/Illumina/manta}, v1.6.0 in 2019-06-25

Current: GitHub, pre-build binary, v1.6.0

License: GPL 3

C++ 87.3\% Python 7.2\% CMake 4.3\% Shell 0.8\% Dockerfile 0.3\% C 0.1\%

Chen, X., Schulz-Trieglaff, O., Shaw, R., Barnes, B., Schlesinger, F., Källberg, M., … Saunders, C. T. (2016). Manta: Rapid detection of structural variants and indels for germline and cancer sequencing applications. Bioinformatics, 32(8), 1220–1222. \url{https://doi.org/10.1093/bioinformatics/btv710}\cb{250}{396}{46}

Updated: 2020-07-21

\section{MuSE}

Official: \url{https://bioinformatics.mdanderson.org/public-software/muse/}

GitHub: \url{https://github.com/danielfan/MuSE}, v1.0-rc in 2016-07-09

Conda: \url{https://anaconda.org/bioconda/muse}, v1.0.rc in 2-19-03

Current: GitHub, pre-build binary, v1.0rc

License: GPL 2

C++ 63.0\% C 36.8\% Makefile 0.2\%

Fan, Y., Xi, L., Hughes, D. S. T., Zhang, J., Zhang, J., Futreal, P. A., … Wang, W. (2016). MuSE: accounting for tumor heterogeneity using a sample-specific error model improves sensitivity and specificity in mutation calling from sequencing data. Genome Biology, 17(1), 178. \url{https://doi.org/10.1186/s13059-016-1029-6}\cb{56}{100}{19}

Updated: 2020-07-21

\section{MuTect1 (deprecated)}

Official: \url{https://software.broadinstitute.org/cancer/cga/mutect}

GitHub: \url{https://github.com/broadinstitute/mutect}, v1.1.6 in 2013-08-30

QBRC: \url{https://github.com/tianshilu/QBRC-Somatic-Pipeline/blob/master/somatic_script/mutect-1.1.7.jar}, v3.1-0-g72492bb (v1.1.7 in filename, v3.1-0-g72492bb in \verb|--version| command) in unknown

Current: QBRC, pre-build binary, v3.1-0-g72492bb

Java 96.3\% Perl 2.3\% Scala 1.4\%

Comment: Java 1.7.

Cibulskis, K., Lawrence, M. S., Carter, S. L., Sivachenko, A., Jaffe, D., Sougnez, C., … Getz, G. (2013). Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples. Nature Biotechnology, 31(3), 213–219. \url{https://doi.org/10.1038/nbt.2514}\cb{1863}{2641}{2003}

Updated: 2020-07-21

\section{Shimmer}

GitHub: \url{https://github.com/nhansen/Shimmer}, v0.2 in 2018-03-07

Current: GitHub, pre-build binary, unknown

License: Public Domain -- United States Government Work

Perl 68.1\% C 11.4\% Python 11.0\% R 7.9\% Other 1.0\% Makefile 0.5\% Shell 0.1\%

Hansen, N. F., Gartner, J. J., Mei, L., Samuels, Y., \& Mullikin, J. C. (2013). Shimmer: Detection of genetic alterations in tumors using next-generation sequence data. Bioinformatics, 29(12), 1498–1503. \url{https://doi.org/10.1093/bioinformatics/btt183}\cb{32}{61}{22}

Updated: 2020-07-21

\section{Strelka2}

Conda: \url{https://anaconda.org/bioconda/strelka}, v2.9.10 in 2018-12

GitHub: \url{https://github.com/Illumina/strelka}, v2.9.10 in 2018-11-07

Current: GitHub, self-build, 2.9.10

License: GPL 3

C++ 81.3\% Python 12.8\% CMake 4.2\% Shell 1.1\% Dockerfile 0.3\% Jupyter Notebook 0.2\% C 0.1\%

Kim, S., Scheffler, K., Halpern, A. L., Bekritsky, M. A., Noh, E., Källberg, M., … Saunders, C. T. (2018). Strelka2: fast and accurate calling of germline and somatic variants. Nature Methods, 15(8), 591–594. \url{https://doi.org/10.1038/s41592-018-0051-x}\cb{32}{146}{3}

Updated: 2020-07-21
\section{VarScan1 (deprecated)}

Official: \url{http://varscan.sourceforge.net/}, v2.3.9 in 2015-06-03, deprecated

Current: N/A

Java

Koboldt, D. C., Chen, K., Wylie, T., Larson, D. E., McLellan, M. D., Mardis, E. R., Weinstock, G. M., Wilson, R. K. and Ding, L. (2009) VarScan: variant detection in massively parallel sequencing of individual and pooled samples., Bioinformatics (Oxford, England), 25(17), pp. 2283–5. doi: 10.1093/bioinformatics/btp373.\cb{687}{1026}{1030}

Updated: 2020-07-21

\section{VarScan2}

Official: \url{http://dkoboldt.github.io/varscan/}

GitHub: \url{https://github.com/dkoboldt/varscan}, v2.4.2 (release) in 2016-05-27, v2.4.4 (file) in 2019-07

Conda: \url{https://anaconda.org/bioconda/varscan}, v2.4.4 in 2019-10-26

Current: GitHub, pre-build binary, v2.4.2

License: The Non-Profit Open Software License version 3.0 (NPOSL-3.0)

Java

Koboldt, D. C., Zhang, Q., Larson, D. E., Shen, D., McLellan, M. D., Lin, L., Miller, C. A., Mardis, E. R., Ding, L. and Wilson, R. K. (2012) VarScan 2: Somatic mutation and copy number alteration discovery in cancer by exome sequencing, Genome Research. Cold Spring Harbor Laboratory Press, 22(3), pp. 568–576. doi: 10.1101/gr.129684.111.\cb{1930}{2823}{2006}

Updated: 2020-07-21

\part{2nd Gen. Variant Caller: CNV, Translocation and other types of SV}
\section{BreakDancer}

Official: \url{http://breakdancer.sourceforge.net/}, v1.1.2\_2013\_03\_08 in 2013-03-06

Official: \url{http://gmt.genome.wustl.edu/packages/breakdancer/}

Conda: \url{https://anaconda.org/bioconda/breakdancer}, v1.4.5 in 2018-04

SorceForge: \url{https://sourceforge.net/projects/breakdancer}, v1.1.2\_2013\_03\_08 in 2013-03-06

GitHub: \url{https://github.com/genome/breakdancer}, archived, v1.4.5 in 2014-08-08

License: GPL v3

Current: N/A

Perl, C++

Chen, K., Wallis, J. W., McLellan, M. D., Larson, D. E., Kalicki, J. M., Pohl, C. S., McGrath, S. D., Wendl, M. C., Zhang, Q., Locke, D. P., Shi, X., Fulton, R. S., Ley, T. J., Wilson, R. K., Ding, L., \& Mardis, E. R. (2009). BreakDancer: An algorithm for high-resolution mapping of genomic structural variation. Nature Methods, 6(9), 677–681. \url{https://doi.org/10.1038/nmeth.1363}\cb{860}{1285}{1582}

Updated: 2020-07-21

\section{BreaKmer}

GitHub: \url{https://github.com/ccgd-profile/BreaKmer}, v0.0.6 in 2020-07-09

Current: N/A

Python 100.0\%

Updated: 2020-07-21

\section{Breakway}

SourceForge: \url{https://sourceforge.net/projects/breakway/}, deprecated, v0.7.1 in 2011-04-01

Perl

Updated: 2020-07-21

\section{ClipCrop}

GitHub: \url{https://github.com/shinout/clipcrop}, ? in 2012-01-27

Current: N/A

JavaScript 100.0\%

Suzuki, S., Yasuda, T., Shiraishi, Y., Miyano, S., \& Nagasaki, M. (2011). ClipCrop: a tool for detecting structural variations with single-base resolution using soft-clipping information. BMC bioinformatics, 12 Suppl 14(Suppl 14), S7. \url{https://doi.org/10.1186/1471-2105-12-S14-S7}\cb{29}{56}{29}

Updated: 2020-07-21

\section{CREST}

GitHub: \url{https://github.com/youngmook/CREST}

Conda: \url{https://anaconda.org/imperial-college-research-computing/crest}, v1.0 in 2018-07

Current: Github, pre-built binary, master

Perl 67.3\% HTML 32.7\%

Updated: 2020-07-21

\section{Delly}

GitHub: \url{https://github.com/dellytools/delly}. v0.8.3 in 2020-03-10

Conda: \url{https://anaconda.org/bioconda/delly}, v0.8.3 in 2020-03-11

Current: GitHub, pre-built binary, v0.8.3

License:  BSD-3-Clause License 

C++ 99.1\% Other 0.9\%

Tobias Rausch, Thomas Zichner, Andreas Schlattl, Adrian M. Stuetz, Vladimir Benes, Jan O. Korbel. DELLY: structural variant discovery by integrated paired-end and split-read analysis. Bioinformatics. 2012 Sep 15;28(18):i333-i339. \url{https://doi.org/10.1093/bioinformatics/bts378}\cb{651}{956}{542}

Updated: 2020-07-21

\section{digit}

GitHub: \url{https://github.com/richard-meier/digit-trl/}, unknown version in 2016-12-06

Current: Github, pre-build biniarirs, 2016-12-06

License: GPL v3

Java 100\%

Meier, R., Graw, S., Beyerlein, P., Koestler, D., Molina, J. R., \& Chien, J. (2017). digit-a tool for detection and identification of genomic interchromosomal translocations. Nucleic Acids Research, 45(9), e72. \url{https://doi.org/10.1093/nar/gkx010}\cb{0}{0}{0}

Updated: 2020-07-21

\section{GASV}

Official: \url{http://compbio.cs.brown.edu/projects/gasv/}

Google: \url{http://code.google.com/p/gasv/downloads/list}

Current: Google, self-built binary, Oct1\_2013

Sindi, S.S., Önal, S., Peng, L.C. et al. An integrative probabilistic model for identification of structural variation in sequencing data. Genome Biol 13, R22 (2012). \url{https://doi.org/10.1186/gb-2012-13-3-r22}\cb{92}{150}{124}

Suzanne Sindi, Elena Helman, Ali Bashir, Benjamin J. Raphael, A geometric approach for classification and comparison of structural variants, Bioinformatics, Volume 25, Issue 12, 15 June 2009, Pages i222–i230, \url{https://doi.org/10.1093/bioinformatics/btp208}\cb{117}{189}{0}

Updated: 2020-07-21

\section{Gustaf}

Official: \url{http://www.seqan.de/apps/gustaf/}

GitHub: \url{https://github.com/seqan/seqan/tree/master/apps/gustaf}

Current: GitHub, self-built binary, v1.0.0 with SeqAn version: 2.4.0

Kathrin Trappe, Anne-Katrin Emde, Hans-Christian Ehrlich, Knut Reinert, Gustaf: Detecting and correctly classifying SVs in the NGS twilight zone, Bioinformatics, Volume 30, Issue 24, 15 December 2014, Pages 3484–3490, \url{https://doi.org/10.1093/bioinformatics/btu431}\cb{28}{53}{20}

Trappe, K. (2012). Multi-Split Mapping of NGS Reads for Variant Detection. Master’s thesis, Freie Universitaet Berlin.\cb{N/A}{N/A}{0}

Updated: 2020-07-21

\section{Hydra-Multi}

GitHub: \url{https://github.com/arq5x/Hydra}, ? in 2014-10-06

C++ 50.3\% Python 45.8\% Shell 3.1\% Perl 0.8\%

Current: N/A

Updated: 2020-07-21

\section{inGAP-SV}

Official: \url{http://ingap.sourceforge.net/}

SourceForge: \url{https://sourceforge.net/projects/ingap/}, v3.1.1 in 2014-04-18

Comment: GUI with no CLI support.

Current: N/A

Java

Qi, J., \& Zhao, F. (2011). inGAP-sv: a novel scheme to identify and visualize structural variation from paired end mapping data. Nucleic acids research, 39(Web Server issue), W567–W575. \url{https://doi.org/10.1093/nar/gkr506}\cb{58}{87}{84}

\section{LUMPY}

GitHub: \url{https://github.com/arq5x/lumpy-sv}, v3.0 in 2019-02-21

Conda: \url{https://anaconda.org/bioconda/lumpy-sv}, v3.0 in 2019-11-28

Current: GitHub, self-built binary, v3.0

License: MIT

Comment: Use Python 2.7

C 76.8\% C++ 20.9\% Python 1.1\% Shell 0.7\% Makefile 0.3\% CMake 0.2\%

Layer, R. M., Chiang, C., Quinlan, A. R., \& Hall, I. M. (2014). LUMPY: A probabilistic framework for structural variant discovery. Genome Biology, 15(6), R84. \url{https://doi.org/10.1186/gb-2014-15-6-r84} \cb{388}{618}{278}

Updated: 2020-07-21

\section{Meerkat}

Official: \url{http://compbio.med.harvard.edu/Meerkat/}, v0.189

Current: N/A

Yang, L., Luquette, L. J., Gehlenborg, N., Xi, R., Haseley, P. S., Hsieh, C. H., Zhang, C., Ren, X., Protopopov, A., Chin, L., Kucherlapati, R., Lee, C., \& Park, P. J. (2013). Diverse mechanisms of somatic structural variations in human cancer genomes. Cell, 153(4), 919–929. \url{https://doi.org/10.1016/j.cell.2013.04.010}\cb{168}{258}{14}

Updated: 2020-07-21

\section{MetaSV}

GitHub: \url{https://github.com/bioinform/metasv}, v0.5.4 in 2017-01-18

Official: \url{bioinform.github.io/metasv/}

Conda: \url{https://anaconda.org/bioconda/metasv}, v0.5.4 in 2018-06

Current: N/A

Comment: Require Python 2.7-2.8

License: BSD-2-Clause License

Python 94.8\% HTML 5.1\% Shell 0.1\%

Mohiyuddin, M., Mu, J. C., Li, J., Bani Asadi, N., Gerstein, M. B., Abyzov, A., Wong, W. H., \& Lam, H. Y. K. (2015). MetaSV: an accurate and integrative structural-variant caller for next generation sequencing. Bioinformatics, 31(16), 2741–2744. \url{https://doi.org/10.1093/bioinformatics/btv204}\cb{47}{83}{40}

Updated: 2020-07-21

\section{PEMer}

Official: \url{http://sv.gersteinlab.org/pemer/}

Current: N/A

Comment: Require Python 2

Korbel, J. O., Abyzov, A., Mu, X. J., Carriero, N., Cayting, P., Zhang, Z., Snyder, M., \& Gerstein, M. B. (2009). PEMer: a computational framework with simulation-based error models for inferring genomic structural variants from massive paired-end sequencing data. Genome biology, 10(2), R23. \url{https://doi.org/10.1186/gb-2009-10-2-r23}\cb{180}{276}{358}

Updated: 2020-07-21

\section{PriSM}

Official: \url{https://www.broadinstitute.org/viral-genomics/prism}

Official: \url{https://software.broadinstitute.org/prism/}

Current: N/A

Perl

Yu, Q., Ryan, E. M., Allen, T. M., Birren, B. W., Henn, M. R., \& Lennon, N. J. (2011). PriSM: a primer selection and matching tool for amplification and sequencing of viral genomes. Bioinformatics (Oxford, England), 27(2), 266–267. \url{https://doi.org/10.1093/bioinformatics/btq624}\cb{9}{13}{8}

Updated: 2020-07-21

\section{Socrates}

Official: \url{http://bioinf.wehi.edu.au/socrates/}

GitHub: \url{https://github.com/PapenfussLab/socrates}, v1.13.1 in 2016-12-15

Current: GitHub, pre-built binary, v1.13.1

License: GPL v3

Java 100.0\%

Schroder, J., Hsu, A., Boyle, S. E., Macintyre, G., Cmero, M., Tothill, R. W., Johnstone, R. W., Shackleton, M., \& Papenfuss, A. T. (2014). Socrates: identification of genomic rearrangements in tumour genomes by re-aligning soft clipped reads. Bioinformatics (Oxford, England), 30(8), 1064–1072. \url{https://doi.org/10.1093/bioinformatics/btt767}\cb{44}{70}{35}

Updated: 2020-07-21

\section{SoftSV}

SourceForge: \url{https://sourceforge.net/projects/softsv/}, v1.4.2 in 2015-09-09

Current: N/A

License: GPL V3

C++

C. Bartenhagen, M. Dugas, Robust and exact structural variation detection with paired-end and soft-clipped alignments: SoftSV compared with eight algorithms, Brief Bioinform. (2015), \url{http://dx.doi.org/10.1093/bib/bbv028}\cb{17}{22}{13}

Updated: 2020-07-21

\section{SVDetect}

Official: \url{http://svdetect.sourceforge.net/Site/Home.html}

SourceForge: \url{ttps://sourceforge.net/projects/svdetect/files/SVDetect}, r0.7m in 2011-07-12

Current: SourceForge, pre-built binary, r0.7m

License: GPL v3

Zeitouni, B., Boeva, V., Janoueix-Lerosey, I., Loeillet, S., Legoix-né, P., Nicolas, A., Delattre, O., \& Barillot, E. (2010). SVDetect: a tool to identify genomic structural variations from paired-end and mate-pair sequencing data. Bioinformatics (Oxford, England), 26(15), 1895–1896. \url{https://doi.org/10.1093/bioinformatics/btq293}\cb{134}{256}{8}

Updated: 2020-07-21

\section{SVMerge}

Official: \url{http://svmerge.sourceforge.net/}

Sourceforge: \url{https://sourceforge.net/projects/svmerge}, v1,2r37 in 2012-08-09

Conda: \url{https://anaconda.org/hcc/svmerge}, v1,2r37 in 2017-05

Current: Sourceforge, pre-built binary, v1,2r37 in 2012-08-09

License: GPL v3

Perl

Wong, K., Keane, T. M., Stalker, J., \& Adams, D. J. (2010). Enhanced structural variant and breakpoint detection using SVMerge by integration of multiple detection methods and local assembly. Genome Biology, 11(12), R128. \url{https://doi.org/10.1186/gb-2010-11-12-r128}\cb{79}{103}{N/A}

Updated: 2020-07-21

\section{SVTools}

GitHub: \url{https://github.com/hall-lab/svtools}, v0.5.1 in 2019-09-13

Conda: \url{https://anaconda.org/bioconda/svtools}, v0.5.1 in 2019-11-18

Conda: \url{https://anaconda.org/bioconda/hall-lab-svtools}, v0.1.1 in 2016-01

Current: Conda, v0.5.1

License: MIT

Python 98.5\% Other 1.5\%

Dave Larson, abelhj, Colby Chiang, Allison Penner Regier, AbhijitBadve, Jim Eldred, … Danny Antaki. (2019, September 12). hall-lab/svtools: svtools v0.5.1 (Version v0.5.1). Zenodo. \url{http://doi.org/10.5281/zenodo.3406745}\cb{N/A}{N/A}{N/A}

Updated: 2020-07-21

\section{ULYSSES}

Official: \url{http://www.lcqb.upmc.fr/ulysses/}

GitHub: \url{https://github.com/gillet/ulysses}, v1.0 in 2017-07-04

Comment: Use Python 2

Current: N/A

License: GPL v3

Python 84.0\% R 15.1\% C 0.9\%

Alexandre Gillet-Markowska, Hugues Richard, Gilles Fischer, Ingrid Lafontaine, Ulysses: accurate detection of low-frequency structural variations in large insert-size sequencing libraries, Bioinformatics, Volume 31, Issue 6, 15 March 2015, Pages 801–808, \url{https://doi.org/10.1093/bioinformatics/btu730}\cb{6}{12}{8}

Updated: 2020-07-21

\part{3rd Gen. Variant Caller: CNV, Translocation and other types of SV}

\section{Sniffles}

GitHub: \url{https://github.com/fritzsedlazeck/Sniffles}, v1.0.11 in 2019-01-30

Conda: \url{https://anaconda.org/bioconda/sniffles}, v1.0.11 in 2019-05

License: MIT

C++ 43.7\% C 30.7\% Makefile 7.9\% Assembly 4.0\% Ada 2.6\% HTML 2.5\% Other 8.6\%

Current: GitHub, self-built binary, v1.0.11

Sedlazeck, F.J., Rescheneder, P., Smolka, M. et al. Accurate detection of complex structural variations using single-molecule sequencing. Nat Methods 15, 461–468 (2018). \url{https://doi.org/10.1038/s41592-018-0001-7}\cb{155}{302}{62}

Updated: 2020-07-21

\part{2nd Gen. VCF/BCF Processor}

\section{BCFTools}

Official: \url{http://www.htslib.org/}

Official: \url{http://samtools.github.io/bcftools/}

GitHub: \url{https://github.com/samtools/bcftools}, v1.10.2 in 2019-12-21

Conda: \url{https://anaconda.org/bioconda/bcftools}, v1.10.2 in 2019-12-22

Current: GitHub, self-build, 1.10.2 with htslib 1.10.2

License: MIT/Expat license or GPL 3

C 87.6\% Perl 9.5\% M4 0.9\% Python 0.7\% Makefile 0.6\% Shell 0.6\% Other 0.1\%

Citation same as Samtools.

Updated: 2020-07-21

\section{CrossMap}

SourceForge: 

Conda: \url{https://anaconda.org/bioconda/crossmap}, v0.33 in 2020-03-26

Official: \url{http://crossmap.sourceforge.net/}, v0.3.8 in 10/09/2019

Current: Conda, v0.33

License: GPL v2

Zhao, H., Sun, Z., Wang, J., Huang, H., Kocher, J.-P., \& Wang, L. (2014). CrossMap: a versatile tool for coordinate conversion between genome assemblies. Bioinformatics (Oxford, England), 30(7), 1006–1007. \url{https://doi.org/10.1093/bioinformatics/btt730}\cb{63}{302}{68}

Updated: 2020-07-21

\section{PLINK}

Official: \url{http://www.cog-genomics.org/plink/2.0/}, v2.00a3LM in 2020-01-15

GitHub: \url{https://github.com/chrchang/plink-ng}

Current: GitHub, pre-built binary, v2.00a3LM in 2020-01-15

License: GPL v3

C 50.6\% C++ 47.5\% Python 1.1\% TeX 0.3\% Makefile 0.2\% Shell 0.1\% Other 0.2\% 

Chang, C. C., Chow, C. C., Tellier, L. C. A. M., Vattikuti, S., Purcell, S. M., \& Lee, J. J. (2015). Second-generation PLINK: rising to the challenge of larger and richer datasets. GigaScience, 4(1). \url{https://doi.org/10.1186/s13742-015-0047-8}\cb{1775}{2870}{1065}

Hill, A., Loh, P.-R., Bharadwaj, R. B., Pons, P., Shang, J., Guinan, E., Lakhani, K., Kilty, I., \& Jelinsky, S. A. (2017). Stepwise Distributed Open Innovation Contests for Software Development: Acceleration of Genome-Wide Association Analysis. GigaScience, 6(5). \url{https://doi.org/10.1093/gigascience/gix009}\cb{9}{12}{4}

Manichaikul, A., Mychaleckyj, J. C., Rich, S. S., Daly, K., Sale, M., \& Chen, W.-M. (2010). Robust relationship inference in genome-wide association studies. Bioinformatics, 26(22), 2867–2873. \url{https://doi.org/10.1093/bioinformatics/btq559}\cb{89}{925}{651}

Galinsky, K. J., Bhatia, G., Loh, P. R., Georgiev, S., Mukherjee, S., Patterson, N. J., \& Price, A. L. (2016). Fast Principal-Component Analysis Reveals Convergent Evolution of ADH1B in Europe and East Asia. American Journal of Human Genetics, 98(3), 456–472. \url{https://doi.org/10.1016/j.ajhg.2015.12.022}\cb{}{169}{}

Yang, J., Lee, S. H., Goddard, M. E., \& Visscher, P. M. (2011). GCTA: A tool for genome-wide complex trait analysis. American Journal of Human Genetics, 88(1), 76–82. {https://doi.org/10.1016/j.ajhg.2010.11.011}\cb{2327}{3549}{3021}

Gaunt, T. R., Rodríguez, S., \& Day, I. N. M. (2007). Cubic exact solutions for the estimation of pairwise haplotype frequencies: Implications for linkage disequilibrium analyses and a web tool ``CubeX.'' BMC Bioinformatics, 8(1), 428. \url{https://doi.org/10.1186/1471-2105-8-428}\cb{200}{247}{7}

Graffelman, J., \& Moreno, V. (2013). The mid p-value in exact tests for Hardy-Weinberg equilibrium. Statistical Applications in Genetics and Molecular Biology, 12(4), 433–448. \url{https://doi.org/10.1515/sagmb-2012-0039}\cb{27}{53}{30}

Graffelman, J., \& Weir, B. S. (2016). Testing for Hardy-Weinberg equilibrium at biallelic genetic markers on the X chromosome. Heredity, 116(6), 558–568. \url{https://doi.org/10.1038/hdy.2016.20}\cb{19}{18}{17}

Wigginton, J. E., Cutler, D. J., \& Abecasis, G. R. (2005). A note on exact tests of Hardy-Weinberg equilibrium. American Journal of Human Genetics, 76(5), 887–893. \url{https://doi.org/10.1086/429864}\cb{942}{1385}{2387}

Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M. A. R., Bender, D., Maller, J., Sklar, P., De Bakker, P. I. W., Daly, M. J., \& Sham, P. C. (2007). PLINK: A tool set for whole-genome association and population-based linkage analyses. American Journal of Human Genetics, 81(3), 559–575. \url{https://doi.org/10.1086/519795}\cb{14485}{20572}{19934}

Updated: 2020-07-21

\section{VCFTools}

SourceForge: \url{http://vcftools.sourceforge.net/}, v0.1.13 in 2015-08-03, branch deprecated

SVN: \url{http://svn.code.sf.net/p/vcftools/code/trunk/}

Official: \url{http://vcftools.github.io/}

GitHub: \url{https://github.com/vcftools/vcftools}, v0.1.16 in 2018-08-03

Current: GitHub, self-build, v0.1.17

Conda: \url{https://anaconda.org/bioconda/vcftools}, v0.1.16 in 2019-05

License: LGPL 3

C++ 44.8\% Perl 44.5\% C 6.6\% Roff 3.0\% Shell 0.8\% M4 0.2\% Makefile 0.1\%

See Samtools: \textbf{VCF Format:}

Updated: 2020-07-21

\part{2nd Gen. VCF Annotation \& Visualization}
\section{ANNOVar}

Official: \url{http://www.openbioinformatics.org/annovar/annovar_download.html}, v2014Dec12

Official: \url{https://doc-openbio.readthedocs.io/projects/annovar/en/latest/user-guide/download/}, v2019Oct24

Current: Official, pre-build binary, \$Date: 2019-10-24 00:05:27 -0400 (Thu, 24 Oct 2019) \$

Wang, K., Li, M., \& Hakonarson, H. (2010). ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Research, 38(16), e164–e164. \url{https://doi.org/10.1093/nar/gkq603}\cb{4862}{6780}{5260}

Updated: 2020-07-21

\section{myVCF}

Official: \url{https://apietrelli.github.io/myVCF/}

ReadTheDocs: \url{https://myvcf.readthedocs.io/en/latest/}

GitHub: \url{https://github.com/apietrelli/myVCF}, v1.0 in 2017-10-09

Current: N/A

License: CC-BY-NC 4.0 International

Python 93.7\% HTML 6.2\% Other 0.1\%

Pietrelli, A. and Valenti, L. (2017) myVCF: a desktop application for high-throughput mutations data management, Bioinformatics, 33(22), pp. 3676–3678. doi: 10.1093/bioinformatics/btx475.\cb{3}{3}{0}

Updated: 2020-07-21
\section{snpEff}

Official: \url{http://snpeff.sourceforge.net/}, v4.3t in 2017-11-24

GitHub: \url{https://github.com/pcingola/SnpEff}, v4.3t in 2017-11-24

Conda: \url{https://anaconda.org/bioconda/snpeff}, v4.5covid19 in 2020-05-07

Current: Official, pre-build binary, 4.3t (build 2017-11-24 10:18)

Comment: Requires Java 1.8 (Java8)

License: LGPL 3

Java 62.0\% HTML 31.5\% Perl 2.2\% Shell 1.7\% CSS 0.8\% Python 0.6\% Other 1.2\%

Cingolani, P., Platts, A., Wang, L. L., Coon, M., Nguyen, T., Wang, L., Land, S. J., Lu, X. and Ruden, D. M. (2012) A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3, Fly. Taylor and Francis Inc., 6(2), pp. 80–92. doi: 10.4161/fly.19695.\cb{3023}{4303}{2797}

Updated: 2020-07-21

\section{snpSift}

Official: \url{http://snpeff.sourceforge.net/}, v4.3t in 2017-11-24

GitHub: \url{https://github.com/pcingola/SnpSift}, v4.3t in 2017-11-24

Conda: \url{https://anaconda.org/bioconda/snpsift}, v4.3.1t in 2020-04-21

Current: Official, pre-build binary, v4.3t (build 2017-11-24 10:18)

License: LGPL 3

Comment: Requires Java 1.8 (Java8)

Java 97.3\% G-code 2.3\% Shell 0.4\%

Cingolani, P., Patel, V. M., Coon, M., Nguyen, T., Land, S. J., Ruden, D. M. and Lu, X. (2012) Using Drosophila melanogaster as a model for genotoxic chemical mutational studies with a new program, SnpSift, Frontiers in Genetics, 3(MAR). doi: 10.3389/fgene.2012.00035.\cb{295}{428}{246}

Updated: 2020-07-21

\section{Variant Effect Predicator}

GitHub \url{https://github.com/Ensembl/ensembl-vep}

Official: \url{https://asia.ensembl.org/info/docs/tools/vep/index.html}, v100.2 in 2020-05-25

Conda: \url{https://anaconda.org/bioconda/ensembl-vep}, v100.2 in 2020-06-01

DockerHub: \url{https://hub.docker.com/r/ensemblorg/ensembl-vep/}, v100.2 in 2020-06

Current: N/A

License: Apache 2.0

Perl 97.6\% Other 2.4\%

McLaren, W., Gil, L., Hunt, S. E., Riat, H. S., Ritchie, G. R. S., Thormann, A., Flicek, P. and Cunningham, F. (2016) The Ensembl Variant Effect Predictor, Genome Biology. BioMed Central Ltd., 17(1), p. 122. doi: 10.1186/s13059-016-0974-4.\cb{1063}{1674}{445}

Updated: 2020-07-21

\section{VarSifter}

GitHub: \url{https://github.com/teerjk/VarSifter}, v1.9 in 2018-08-01

Official: \url{https://research.nhgri.nih.gov/software/VarSifter/}, branch deprecated

FTP: \url{ftp://ftp.nhgri.nih.gov/pub/software/VarSifter/}, v1.7 in 2015-01-08

Current: N/A

License: THE JUNG LICENSE (Another kind of BSD)

Java 91.1\% HTML 8.4\% Makefile 0.5\%

Teer, J. K., Green, E. D., Mullikin, J. C., \& Biesecker, L. G. (2012). VarSifter: Visualizing and analyzing exome-scale sequence variation data on a desktop computer. Bioinformatics, 28(4), 599–600. \url{https://doi.org/10.1093/bioinformatics/btr711}\cb{102}{174}{N/A}

Updated: 2020-07-21

\part{2nd Gen. SV Annotation \& Visualization}

\section{Ciros}

Official: \url{http://circos.ca/}, v0.69-9  in 2019-07-03

Current: v0.69-9

License: GPL V3

Perl

Krzywinski, M., Schein, J., Birol, I., Connors, J., Gascoyne, R., Horsman, D., Jones, S. J., \& Marra, M. A. (2009). Circos: An information aesthetic for comparative genomics. Genome Research, 19(9), 1639–1645. \url{https://doi.org/10.1101/gr.092759.109 }\cb{4131}{5568}{5767}

Updated: 2020-07-21

\section{SVTyper}

GitHub: \url{https://github.com/hall-lab/svtyper}, v0.7.1.in 2019-09-05

Conda: \url{https://anaconda.org/bioconda/svtyper}, v0.7.1 in 2019-10-04

Current: Conda, v0.1.4

License: MIT

Python 98.8\% Other 1.2\%

Same as speedseq.

Updated: 2020-07-23

\part{2nd Gen. Pipelines}
\section{SpeedSeq}

GitHub: \url{https://github.com/hall-lab/speedseq}, v0.1.2 in 2017-01-10

Current: GitHub, self-built binary, v0.1.2

License: The MIT License (MIT)

C 75.5\% Roff 8.4\% Perl 8.0\% Shell 3.1\% Makefile 1.5\% C++ 1.0\% Other 2.5\%

Chiang, C., Layer, R. M., Faust, G. G., Lindberg, M. R., Rose, D. B., Garrison, E. P., Marth, G. T., Quinlan, A. R. and Hall, I. M. (2015) SpeedSeq: Ultra-fast personal genome analysis and interpretation, Nature Methods. Nature Publishing Group, 12(10), pp. 966–968. doi: 10.1038/nmeth.3505.\cb{133}{216}{29}

Updated: 2020-07-21

\part{R, IDE and R Packages}

\section{Microsoft Open-R}

Official: \url{https://mran.microsoft.com/}, v3.5.3 in 2019-05-17

GitHub: \url{https://github.com/microsoft/microsoft-r-open}

Conda: \url{https://anaconda.org/r/mro-base}, v3.5.1 in 2020-02-12

Current: N/A

License: GPL V2, V3

R 36.6\% C 30.6\% Fortran 23.3\% HTML 2.9\% Shell 1.2\% M4 1.1\% Other 4.3\%

Updated: 2020-07-21

\section{R}

Conda: \url{https://anaconda.org/r/r-base}, v3.6.1 in 2020-05-20

Official: \url{https://www.r-project.org/}, v4.0.2 in 2020-06-22

SVN: \url{https://svn.r-project.org/R-dev-web/trunk/index.html}

Current: Conda, v4.0.2

License: GPL 2

R Core Team (2020) R: A Language and Environment for Statistical Computing. Vienna, Austria. Available at: \url{https://www.r-project.org}.

Updated: 2020-07-21

\section{RStudio}

Official: \url{https://rstudio.com/}, v1.3 in 2020-05-27

Conda: \url{https://anaconda.org/r/rstudio}, v1.1.456 in 2020-03-10

GitHub: \url{https://github.com/rstudio/rstudio}, v1.3.959 in 2020-05-27

Current: N/A

License: Commercial or AGPL 3

Java 36.1\% C++ 31/3\% JavaScript 20.4\% R 3.3\% C 3.2\% TypeScript 2.6\% Other 3.1\% 

RStudio Team (2015). RStudio: Integrated Development for R. RStudio, Inc., Boston, MA URL http://www.rstudio.com/.

Updated: 2020-07-21





\part{Python and Python Packages}

\section{Python3}

Conda: \url{https://anaconda.org/conda-forge/python}, v3.8.3 in 2020-06-02

Official: \url{https://www.python.org/}, v3.8.3 in 2020-05-13

GitHub: \url{https://github.com/python/cpython}, v3.9.0b3 in 2020-06-10

Current: Conda, v3.7.6

License: PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2

Python 63.0\% C 30.1\% Objective-C 4.2\% C++ 1.3\% HTML 0.4\% M4 0.4\% Other 0.6\% 

\textbf{From General Python 3 FAQ: \url{https://docs.python.org/3/faq/general.html}}

Are there any published articles about Python that I can reference?

It's probably best to cite your favourite book about Python.

The very first article about Python was written in 1991 and is now quite outdated.

Rossum, G. van and Boer, J. de (1991) Interactively Testing Remote Servers Using the Python Programming Language, CWI Quarterly, pp. 283–303.

Updated: 2020-07-21

\section{pyBedTools}

Conda: \url{https://anaconda.org/bioconda/pybedtools}, v0.8.1 in 2020-05-10

GitHub: \url{https://github.com/daler/pybedtools}, v0.8.0 in 2018-11-25

Official: \url{http://daler.github.io/pybedtools/}

Current: N/A

License: GPL v2

Python 90.5\% C++ 7.5\% Shell 1.5\% Other 0.5\%

Updated: 2020-07-21

\part{File Specifications}
\section{FASTQ}

Rhizobium, G. E. (2013) Complete Genome Sequence of the Sesbania Symbiont and Rice, Nucleic acids research. Oxford Academic, 1(1256879), pp. 13–14. doi: 10.1093/nar.\cb{22}{25}{13}

Updated: 2020-07-21


\section{SAM, BAM, CRAM, BCF, VCF, CSI, BAI}

GitHub: \url{https://github.com/samtools/hts-specs}

Official: \url{http://samtools.github.io/hts-specs/}

TeX 97.8\% HTML 1.3\% Makefile 0.3\% Shell 0.2\% CSS 0.2\% JavaScript 0.1\% Dockerfile 0.1\% 

Sequence Alignment/Map Format (SAM) v1 in 2020-04-30

CRAM format v2.1 in 2019-01-22, Apache license 2.0

CRAM format v3 in 2020-01-20, Apache license 2.0

BCF v1 in 2019-11-20, deprecated

BCF v2.1 in 2019-07-24

CSI v1 in 2019-11-20

Tabix index file format in 2018-07-06

Variant Call Format (VCF) v4.1 in 2020-03-03

Variant Call Format (VCF) v4.2 in 2020-03-03

Variant Call Format (VCF) v4.3 in 3030-05-27

GA4GH File Encryption Standard (crypt4gh) in 2019-10-21

Htsget retrieval API spec v1.2.0 in 2020-02-20

Refget API Specification v1.0.0 in 2020-05-09

Updated: 2020-07-21

\part{How We Install Software}
\section{Preface}
Imagine this: You're building your DNA-Seq analysis pipeline due to some published pipelines, which asked you to install several software. However, you feel it hard because you're on an obsolete CentOS with obsolete GCC, GLibC and BinUtils. You do not have access to root and Docker privilege, either. What will you do? In this part, I'll share yu with some of my experience in installing fantastic software mentioned in this book. Wish you a pleasant journey!

\section{Assumptions}
We assume that you're under an obsolete CentOS with obsolete GCC, GLibC and BinUtils, and do not have access to root and Docker privilege.

Please set up the installation directories before we continue. The reason of performing this step is that almost all C \& CPP programs using GNU Make or CMake assumes that you have root privilege by default, which is not true under current circumstance. So, we need to direct the GNU Make program to install to an non-standard installation path, like \verb|$HOME/usr/local|.

This step only needed to be performed once per machine. Once executing following two lines of codes, there's no need to worry about other directories. They'll be created automatically during installation.

\verb|mkdir -p $HOME/bin $HOME/bin/bio.d $HOME/bin/bio.d/JAR|

\verb|mkdir -p $HOME/usr $HOME/usr/bup $HOME/usr/src $HOME/usr/src/TXZ|

The organizing structure we'll finally see is as follows (\verb|->| means soft link and can be created by \verb|ln -s $SOURCE $DEST|):

\begin{verbatim}
$HOME
|-bin Binaries pre-compiled.
| |-bio.d Bioinformatics biniaries.
| | |-JAR Bioinformatics Java executbles.
|-usr
| |-bin -> local/bin Directories used to hold some programs with instandard PATH.
| |-bup The backup directory.
| |-include -> local/include
| |-local Our installation prefix.
| | |-bin Binaries compiled.
| | |-doc Documents that will never be used.
| | |-etc Configuration files.
| | |-include C&CPP INCLUDE path.
| | |-info Documentations written in GNU Info.
| | |-jar JAVA executables.
| | |-lib Shared libraries.
| | | |-pkgconfig Diectory for storing pc files.
| | |-lib64 Another directory for shared libraries.
| | | |-pkgconfig Another diectory for storing pc files.
| | |-libexec Executables that will not appear in $PATH, but used by other programs.
| | |-man Documentations written in Roff MAN.
| | |-sbin Where we install software that requires root previledge originally,
| | | like lighttpd. 
| | |-share Shared files for various purpose, like common translations.
| | |-x86_64-pc-linux-gnu/ Files for GNU BinUtils.
| |-sbin -> local/sbin Same as $HOME/usr/bin
| |-share -> local/share Same as $HOME/usr/bin
| |-src Directory for storing source codes cloned from Git hosting providers like GitHub.
| | |-TXZ Directory for storing source codes in tar.xz format.
\end{verbatim}

\section{DO NOT Rely Everything on Conda}
If you learned how to use Python, You might be familiar with Anaconda, a package manager for Python. As you may see, you can install lots of software in Conda by adding a channel named ``Bioconda''. 

The reason I do not suggest you rely everything on Conda is that Conda will pollute the environment variables by default. For example, Conda will pollute \verb|$PERL5LIBS|, making your Perl programs fail; Conda will pollute \verb|C_INCLUDE_PATH|, making you unable to compile programs written in C; Conda will pollute \verb|LD_LIBRARY_PATH|, making self-compiled C programs fail.

\section[First Things First]{First Things First: Some Conventions; How We Compile C \& CPP Programs using GNU Make; and How To Update GCC}
In the following lines, we'll teach you how to install a C \& CPP program based on GNU Make.

You need to make sure that you have an active C \& CPP compiler on your machine, such as \verb|gcc|, \verb|g++| or \verb|clang| and \verb|clang++|. You may check this by \verb|which $PROGRAM_NAME|, where \verb|$PROGRAM_NAME| is the program name you wish to test. e.g. for \verb|gcc|:
\begin{verbatim}
$ which gcc
~/usr/local/bin/gcc
\end{verbatim}
OK, you got your GCC at \verb|~/usr/local/bin/gcc|.

Or, for program \verb|foo|:
\begin{verbatim}
$ which foo
/usr/bin/which: no foo in (/usr/bin:/bin:/sbin:/usr/sbin:/mnt/sysimage/bin:/mnt/sysimage/usr/bin:/mnt/sysimage/usr/sbin:/mnt/sysimage/sbin:/sbin:/usr/sbin:)
\end{verbatim}

The system failed to find program \verb|foo|.

If you do not have any active C \& CPP compiler on your machine, contact your system administrator (If they do not want to install them, change your computer). This is {\color{red}FATAL}.

Turn off Conda by commenting all Conda scripts at your \verb|.bashrc| and restart your \verb|bash|. This will stop Conda from polluting your environment variables--One of the major reasons of GCC compilation failure.

\subsection{A Sample C Program using GNU Make}

In this section, we'll use GNU MPFR Library as our example. You do not need to download and install this program, just see how we deal with errors emerged.

\begin{enumerate}
\item Downloading the files: Use \verb|curl| or \verb|wget| (You should have at least one of them) or \verb|axel| (You can install it to accelerate downloading speed) to retrieve the newest version of \verb|mpfr|:

e.g. \verb|wget https://www.mpfr.org/mpfr-current/mpfr-4.1.0.tar.xz|

or, \verb|axel https://www.mpfr.org/mpfr-current/mpfr-4.1.0.tar.xz|

or, \verb|curl https://www.mpfr.org/mpfr-current/mpfr-4.1.0.tar.xz -o mpfr-4.1.0.tar.xz|

\item Extract the archive by GNU Tar or YuZJLab AutoUnzip \footnote{I'm very glad to advertise my own products.}.

e.g. \verb|tar -xJvf mpfr-4.1.0.tar.xz| or \verb|autounzp mpfr-4.1.0.tar.xz| will extract \verb|mpfr-4.1.0.tar.xz| to directory \verb|mpfr-4.1.0.tar.xz|. \verb|cd mpfr-4.1.0|

\item Execute \verb|./configure| with option \verb|--prefix=$HOME/user/local|.

\verb|./configure --prefix=$HOME/user/local|

Useful \& Common options:
\begin{enumerate}
\item \verb|--prefix=$INSTALLATION_PREFIX| Install this program in \verb|$INSTALLATION_PREFIX|. e.g put executables in \verb|$INSTALLATION_PREFIX/bin|, shared libraries in \verb|$INSTALLATION_PREFIX/lib|, Roff MAN pages in \verb|$INSTALLATION_PREFIX/man|, etc.
\item \verb|--help| Give out this help.
\item \verb|--enable-FEATURE| or \verb|--enable-FEATURE=yes| Enable some \verb|FEATURE|.
\item \verb|--disable-FEATURE| or \verb|--enable-FEATURE=no| Disable some \verb|FEATURE|.
\item \verb|--with-PACKAGE| or \verb|--with-PACKAGE=$PAKAGE_PATH| Build with \verb|PACKAGE| (in \verb|$PAKAGE_PATH|).
\item \verb|--without-PACKAGE| or \verb|--with-no-PACKAGE| or \verb|--with-PACKAGE=no| Build without \verb|PACKAGE|.
\end{enumerate}
Useful \& Common arguments:
\begin{enumerate}
\item \verb|CC=$C_CMPILER_PATH| Use \verb|$C_CMPILER_PATH| as C Compiler, can be executable name or executable path.

e.g, configure a program with C compiler \verb|clang|:

\verb|./configure CC=clang| or \verb|./configure CC=$(which clang)|

or,\verb|./configure CC=$HOME/usr/local/bin/clang|.

\item \verb|CFLAGS=$C_COMPILER_FLAGS| defines C compiler flags.

e.g. compile with \verb|libiconv|:

\verb|./configure CFLAGS='-liconv'|

or, \verb|./configure CC='gcc -liconv'|

\item \verb|CXX=$CPP_CMPILER_PATH| and \verb|CXXFLAGS=$CPP_COMPILER_FLAGS| are similar with those in \verb|CC| and \verb|CCFLAGS|.

\item \verb|AR=$AR_PATH| and \verb|LD=$LD_PATH| can be executable name or executable path for \verb|ar| and \verb|ld|. \verb|LDFLAGS=$LD_FLAGS| are those arguments passed to linker \verb|ld|.
\end{enumerate}

Please be aware that for arguments, you can also set them as environment variables \footnote{This is how a Conda pollute your environment!} e.g. You can add \verb|export CC=clang| to \verb|$HOME/.bashrc| to use \verb|clang| as C compiler by default. You can also use them before \verb|.configure|.

e.g. \verb|./configure CC=clang|

equals to \verb|CC=clang ./configure|

or

\begin{verbatim}
CC=clang
./configure
\end{verbatim}

e.g, configure a program with compiler \verb|clang|, customized linker and library \verb|libiconv|:

\verb|./configure --prefix=$HOME/user/local CC=clang CXX=clang++ LD=/usr/bin/ld LDFLAGS='-liconv'|

\item \verb|make && make install|. \verb|make| will compile the program while \verb|make install| will install them. Use option \verb|-j| to use multiple steps. E.g. \verb|make -j20|

It is easy to understand the \verb|make install| step: Copy the files, create links and modify some system configurations. This makes \verb|make install| one of the easiest step. However, \verb|make| is more complicated and lots of problems emerges here. 
\end{enumerate}
\subsection{How C Compiler Compiles Executables}
For C code named \verb|hello.c| like this:
\begin{verbatim}
#include <stdio.h>
int main()
{
    printf("Hello, World!");
    return 0;
}
\end{verbatim}

Can be compiled like this: \verb|gcc -o hw hello.c|. This includes:

\subsubsection{Pre-processing of the source code}

This step includes finding and attaching header files (\verb|.h| \& \verb|.hpp|).

CMD: \verb|gcc -E hello.c > hello_pp.c|

The compiler (\verb|gcc| here) will find the line \verb|#include <stdio.h>| which asked for a header file named \verb|stdio.h| in an environment variable named \verb|C_INCLUDE_PATH| (\verb|CPLUS_INCLUDE_PATH| for CPP) which is now setted to \verb|$HOME/usr/local/include|. The compiler will search system INCLUDE path and \verb|C_INCLUDE_PATH| for \verb|stdio.h|, which appears to be in \verb|/usr/include|. All path searched can be viewed by adding option \verb|-v| (which means ``verbose''), e.g.
\begin{verbatim}
ignoring duplicate directory "$HONE/usr/local/include"
#include "..." search starts here:
#include <...> search starts here:
$HONE/usr/local/include
.
$HONE/usr/local/lib/gcc/x86_64-pc-linux-gnu/10.1.0/include
/usr/local/include
$HONE/usr/local/lib/gcc/x86_64-pc-linux-gnu/10.1.0/include-fixed
/usr/include
End of search list.
\end{verbatim}

An error will be reported if a header file was not found. e.g. If I change \verb|stdio.h| to \verb|aastdio.h| in \verb|hello.c|, there will be:

\begin{verbatim}
hello.c:1:10: fatal error: aastdio.h: No such file or directory
1 | #include <aastdio.h>
  |          ^~~~~~~~~~~
compilation terminated.
\end{verbatim}

In this situation, you should check:

\begin{enumerate}
\item Whether this header file exists in your \verb|C_INCLUDE_PATH|.
\item Whether this header file exists in a directory (\verb|$DIR|) not included in your \verb|C_INCLUDE_PATH|. If so, add it to \verb|C_INCLUDE_PATH| or use \verb|-I$DIR|.

e.g. I created \verb|aastdio.h| in \verb|./inc/| by \verb|mkdir inc;ln /usr/include/stdio.h inc/aastdio.h -s|. I can compile this code by \verb|gcc -o hw hellow.c -I./inc|.
\item Whether this header file can be aliased. If so, use soft links. e.g. Sometimes some program will rely on \verb|curses.h|, which is currently \verb|ncurses/ncurses.h|:

\verb|ln -s $HOME/usr/local/include/ncurses/ncurses.h $HOME/usr/local/include/curses.h|
\end{enumerate}
If no above situation suits, you should search where this header come from. Maybe it's time for installing a new library.

Another potential problem is that the C compiler will search for the first header file in \verb|C_INCLUDE_PATH|. So, if you're using Conda, Conda will add its own \verb|include| folder to te top of \verb|C_INCLUDE_PATH|. Lots of problem will emerge if Conda do not have a correct version of the file you want.

If you want to see the results produced by pre-processing step, you can add option \verb|-E|. e.g.
\begin{verbatim}
$ gcc -E hellow.c -I./inc|grep -v '^#'|grep -v ^$
typedef long unsigned int size_t;
typedef unsigned char __u_char;
typedef unsigned short int __u_short;
typedef unsigned int __u_int;
typedef unsigned long int __u_long;
typedef signed char __int8_t;
typedef unsigned char __uint8_t;
[...]
extern void funlockfile (FILE *__stream)
__attribute__ ((__nothrow__ , __leaf__))
;
int main()
{
    printf("Hello, World!");
    return 0;
}
\end{verbatim}
\subsubsection{Compile the pre-processed file into assembly code}

CMD: \verb|gcc -S hello_pp.c|

or, \verb|gcc -S hello.c|.

The assembly code file is like:

\begin{verbatim}
        .file   "hello_pp.c"
        .text
        .section        .rodata
.LC0:
        .string "Hello, World!"
        .text
        .globl  main
        .type   main, @function
main:
.LFB0:
        .cfi_startproc
        pushq   %rbp
        .cfi_def_cfa_offset 16
        .cfi_offset 6, -16
        movq    %rsp, %rbp
        .cfi_def_cfa_register 6
        movl    $.LC0, %edi
        movl    $0, %eax
        call    printf
        movl    $0, %eax
        popq    %rbp
        .cfi_def_cfa 7, 8
        ret
        .cfi_endproc
.LFE0:
        .size   main, .-main
        .ident  "GCC: (GNU) 10.1.0"
        .section        .note.GNU-stack,"",@progbits
\end{verbatim}
\subsubsection{Compile the assembly code into object code}

CMD: \verb|as hello_pp.s -o hello.o|

or, \verb|gcc -c hello_pp.c -o hello.o|

or, \verb|gcc -c hello.c -o hello.o|.

In this step, assembly code into object code, which is binary. This binary have data segment, text (code) segment and a symbol table which stores all symbols defined and used.

We can use command \verb|nm| to see symbols inside a object file.

\subsubsection{Linking object code into machine code (Executables)}

This step links several object file to an executable.

CMD: \verb|gcc hello.o -o hello|

Let's use a more complicated example:

\verb|add.c| includes:
\begin{verbatim}
int add(int a,int b)
{
    return a+b;
}
\end{verbatim}

\verb|math.c| includes:
\begin{verbatim}
#include <stdio.h>
int b=1;
int c=2;
int main()
{
    int a;
    a=add(b,c);
    printf("%d",a);
    return 0;
}
\end{verbatim}

Command \verb|gcc math.c add.c -o math| will generate executable \verb|math|\footnote{When we have lots of files, we use:\\{\tt gcc -c math.c add.c\\\# Previous line generates math.o add.o\\gcc math.o add.o -o math}\par Thus, if there's any error emerged in any {\tt .c} files, we just need to edit and re-compile this file only.}. Command \verb|gcc -c math.c -o math.o| will generate \verb|math.o| without error. However, if we use \verb|gcc math.c|, we will find an error:

\begin{verbatim}
ld: /tmp/cc7gkfN8.o: in function `main':
math.c:(.text+0x18): undefined reference to `add'
collect2: error: ld returned 1 exit status
\end{verbatim}

This means that the linker failed to work. Why? Let's have a look at those object files (Words after \verb|#| are added by me):

\begin{verbatim}
$ nm math.o
                 U add      # Undefined.
0000000000000000 D b        # Defined data.
0000000000000004 D c        # Defined data.
0000000000000000 T main     # Defined code.
                 U printf   # Unefined.
$ nm add.o
0000000000000000 T add      # Defined code.
\end{verbatim}

After encountering fiction \verb|add|, \verb|ld| the linker will start finding where this function is from, which cannot be found because we do not provide \verb|add.o| or any other object file defining function \verb|add| in its symbol table.

Linking to static library (\verb|.a|) includes:

\begin{enumerate}
\item Search environment variables \verb|LIBRARY_PATH|.
\item Extract the library into object files;
\item Get the object files we need;
\item Merge them together. For executables, symbol table will be deprecated.
\end{enumerate}

Note: when linking libraries, \verb|ld| by itself does not look for libraries in either \verb|LIBRARY_PATH| or \verb|LD_LIBRARY_PATH|. It's only when \verb|gcc| invokes \verb|ld| that \verb|LIBRARY_PATH| becomes used. When using stand-alone \verb|ld|, you should use option \verb|-L|. e.g. \verb|ld -L$HOME/usr/local/lib|

They're all done at compilation time. So, there should be no error when executing executables.

Linking to dynamic library (\verb|.so|) includes:

\begin{enumerate}
\item Search environment variables \verb|LIBRARY_PATH| and write library information to executables, done at compilation time.
\item Search environment variables \verb|LD_LIBRARY_PATH| and load them at load time \& run time.
\end{enumerate}

So, if you're encountering a error like this \footnote{Another error caused by Conda. This error emerges because I compiled R when Conda is activated (which modified {\tt LIBRARY\_PATH} and {\tt LD\_LIBRARY\_PATH}) and want to use R when Conda is deactivated.}:

R: error while loading shared libraries: libicuuc.so.64: cannot open shared object file: No such file or directory

You may re-compile R. You'd better deactivate Conda when compiling software.

\subsection{Non-Standard GNU Make Programs Without configure}
\begin{enumerate}
\item Please check whether there's \verb|bootstrap.sh| or \verb|autogen.sh|. If so, you may execute them and a \verb|configure| file will be generated.
\item If there's \verb|configure.ac| or \verb|configure.in|, please execute \verb|autopoint -ifv &&  autoreconf -ifv| and a \verb|configure| file will be generated.
\item If there's \verb|Makefile| only and \verb|install| is defined, please use \verb|PREFIX=$HOME/usr/local| or \verb|DESTDIR=$HOME| as arguments for \verb|make install|. If not, skip \verb|make install|. Copy the binary produced directly to \verb|$HOME/bin| or \verb|$HOME/bin/bio.d|.
\end{enumerate}

\subsection{Skip Makeinfo Error}
Execute \verb|make| with \verb|MAKEINFO='bash -c exit 0'|

\subsection{Installing GCC}
\begin{enumerate}
\item \verb|unset LIBRARY_PATH CPATH C_INCLUDE_PATH PKG_CONFIG_PATH CPLUS_INCLUDE_PATH INCLUDE| to avoid errors during \verb|make|.

\item Download and extract GCC source code.

\item Note that you need to add option \verb|--with-gmp=$HOME/user/local| when executing \verb|./configure|.

\item \verb|make| (which will take several hours if you only use one thread) and \verb|make install|.
\end{enumerate}

Other programs like GNU BinUtils \footnote{Utilities used in compiling C \& CPP codes, like {\tt ld} or {\tt ar}. Available from \url{https://www.gnu.org/software/binutils/}.}, GNU TeXInfo \footnote{Lots of GNU programs write their documentations in GNU TeXInfo (a language). This software contains {\tt makeinfo} which can transcribe GNU TeXInfo ({\tt .texi}) to GNU Info ({\tt .info}), Roff MAN ({\tt .1}, {\tt .2}, etc.), PDF and HTML. Available from \url{https://www.gnu.org/software/texinfo/}.}, GNU M4 \footnote{A dependence for GNU Autotools. Available from \url{https://www.gnu.org/software/m4/}.}, GNU Autotools \footnote{Introduced before. Availble from \url{https://www.gnu.org/software/autoconf/}, \url{https://www.gnu.org/software/autogen/} and \url{https://www.gnu.org/software/automake/}.}, GNU CoreUtils \footnote{CoreUtils like {\tt ls}, {\tt mkdir} or {\tt rm}. Available from \url{https://www.gnu.org/software/coreutils/}.}, GNU Bash\footnote{Some programs requires newer versions of Bash, like YuZJLab LinuxMiniPrograms. Available from \url{https://www.gnu.org/software/bash/}.} can (and should) be installed in a similar way.

\section{Special C Programs}

\subsection{Ncurses}
\verb|./configure --prefix=$HOME/usr/local LDFLAGS='-liconv' --enable-widec --with-shared|

After installation:

\begin{verbatim}
cd $HOME/usr/local/lib
ln -s libncurses.so libncursesw.so
ln -s libncursesw.so libncurses.so
ln -s libncurses.so.6 libncursesw.so.6
ln -s libncursesw.so.6 libncurses.so.6
ln -s libncursesw.so.6.2 libncurses.so.6.2
ln -s libncurses.so libncursesw.so
ln -s libncursesw.so libncurses.so
ln -s libncurses.so.6 libncursesw.so.6
ln -s libncursesw.so.6 libncurses.so.6
ln -s libncursesw.so.6.2 libncurses.so.6.2
\end{verbatim}

\subsection{R}
\begin{verbatim}
./configure --prefix=$HOME/usr/local --with-blas\
 --with-lapack --with-tcltk --enable-R-shlib --with-x=no
\end{verbatim}

\section{Java Programs}

\subsection{Binary}
We just download JAR executables to \verb|$HOME/bin/bio.d/JAR| and add a wrapper like this:

\begin{verbatim}
#!/usr/bin/env bash
${HOME}/bin/jdk1.7.0_80/bin/java -Xmx31g -jar $(dirname ${0})/mutect-1.1.7.jar ${@}
\end{verbatim}

\part{Fantastic Configurations}
In this part, you'll see some fantastic configurations we used to accelerate our performance. All mirror sites have been pointed to Chinese mirrors.

\section{.bashrc}
(This title should not be visible.)
\begin{verbatim}
# Source global definitions
if [ -f /etc/bashrc ]; then
. /etc/bashrc
fi

# Git helper, can be found at git's source code.
. ~/.git-prompt.sh

# Fantastic $PS1 with additional linebreak.
function __prevp (){
local r=${?}
if [ ${r} -eq 0 ];then
echo -e "\e[42m\e[30m${r}\e[32m\e[46m \e[0m"
else
echo -e "\e[41m\e[30m${r}\e[31m\e[46m \e[0m"
fi
}
PS1='$(__prevp)\e[30m\e[46m$(date +%Y-%m-%d_%H-%M-%S)\e[43m\e[36m \e[30m${PWD}
\e[42m\e[33m \e[30m$(__git_ps1 " (%s)")\e[32m\e[40m \e[0m\n\$ '


# Conda setup scripts omitted.

export CMAKE_PREFIX_PATH="$HOME/usr/local/CMAKE_PREFIX:$CMAKE_PREFIX_PATH"

export PERL5LIB="$HOME/usr/local/lib/perl5/site_perl/5.30.1/x86_64-linux/:
$HOME/usr/local/lib/perl5/site_perl/5.30.1/:$HOME/bin/cpan.d/lib/perl5/"

export PKG_CONFIG_PATH="$HOME/usr/local/lib/pkgconfig/:
$HOME/usr/local/lib64/pkgconfig/:$PKG_CONFIG_PATH"

export PKG_CONFIG_LIBFIR="$HOME/usr/local/lib64/pkgconfig/:
$HOME/usr/local/lib/pkgconfig/:$PKG_CONFIG_LIBFIR"

export MANPATH="$HOME/bin/cpan.d/man/man3:$HOME/usr/local/share/man:$MANPATH"
export INFOPATH="$HOME/usr/local/share/info:$INFOPATH"
export LD_LIBRARY_PATH="$HOME/usr/local/lib/:$HOME/usr/local/lib64/:$LD_LIBRARY_PATH"
export LIBRARY_PATH="$HOME/usr/local/lib/:$HOME/usr/local/lib64/:$LIBRARY_PATH"
export LD_RUN_PATH="$HOME/usr/local/lib:$HOME/usr/local/lib64/:$LD_RUN_PATH"
export CPLUS_INCLUDE_PATH="$HOME/usr/local/include:$CPLUS_INCLUDE_PATH"
export C_INCLUDE_PATH="$HOME/usr/local/include:$C_INCLUDE_PATH"
alias du="du -h" # More readable `du`.
alias df="df -h" # More readable `df`.
alias diff="diff -u" # Make the output of `diff` similiar to `git diff`.
alias ls="ls -lhF --color=auto" # More readable ls.
alias shutdown="echo What the hell you\'re thinking?\!" # [Palmface]
alias reboot="echo What the hell you\'re thinking?\!"
alias sudo="echo What the hell you\'re thinking?\!"
alias rm="rm -i" # Safer `rm`.
alias git-log="git log --pretty=oneline --abbrev-commit --graph --branches"
# More readable `git log`.
alias top="htop" # htop is a task manager better than top.
alias emacs="emacs -nw"
#OLDPATH
export PATH="$HOME/bin/apache-ant-1.9.15/bin:$HOME/bin/go.git/bin:$BIOD:$PATH"
#BIO.D
export BIOD="$HOME/bin/bio.d" # Good alias!
export PATH="$PATH:$BIOD/blat:$BIOD/CAP3:$BIOD/JAR:$BIOD/circos-0.69-9/bin:$BIOD/speedseq.git/bin"
#LOCAL
export PATH="$HOME/usr/local/bin:$PATH"
# I splitted the $PATH variables to make it more readable.
\end{verbatim}

\section{.condarc}
\begin{verbatim}
channels:
- https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda/
- https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge/
- https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
- https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
report_errors: false
\end{verbatim}

\section{.emacs}
\begin{verbatim}
(package-initialize)
(custom-set-variables)
(global-linum-mode 1) ;
(setq linum-format "%d ");
(setq backup-directory-alist (quote (("." . "~/.emacs-backups"))))
\end{verbatim}

\section{.Rprofile}
\begin{verbatim}
options("repos" = c(CRAN="https://mirrors.tuna.tsinghua.edu.cn/CRAN/"))
\end{verbatim}

\end{document}
